{"matches":[{"vulnerability":{"id":"GHSA-j7hp-h8jx-5ppr","dataSource":"https://github.com/advisories/GHSA-j7hp-h8jx-5ppr","namespace":"github:language:python","severity":"High","urls":[],"description":"libwebp: OOB write in BuildHuffmanTable","cvss":[{"type":"Secondary","version":"3.1","vector":"CVSS:3.1/AV:N/AC:L/PR:N/UI:R/S:U/C:H/I:H/A:H","metrics":{"baseScore":8.8,"exploitabilityScore":2.9,"impactScore":5.9},"vendorMetadata":{}}],"knownExploited":[{"cve":"CVE-2023-4863","vendorProject":"Google","product":"Chromium WebP","dateAdded":"2023-09-13","requiredAction":"Apply mitigations per vendor instructions or discontinue use of the product if mitigations are unavailable.","dueDate":"2023-10-04","knownRansomwareCampaignUse":"unknown","urls":["https://chromereleases.googleblog.com/2023/09/stable-channel-update-for-desktop_11.html?m=1","https://nvd.nist.gov/vuln/detail/CVE-2023-4863"],"cwes":["CWE-787"]}],"epss":[{"cve":"CVE-2023-4863","epss":0.93991,"percentile":0.9988,"date":"2025-06-02"}],"fix":{"versions":["10.0.1"],"state":"fixed"},"advisories":[],"risk":85.57500000000002},"relatedVulnerabilities":[{"id":"CVE-2023-4863","dataSource":"https://nvd.nist.gov/vuln/detail/CVE-2023-4863","namespace":"nvd:cpe","severity":"High","urls":["http://www.openwall.com/lists/oss-security/2023/09/21/4","http://www.openwall.com/lists/oss-security/2023/09/22/1","http://www.openwall.com/lists/oss-security/2023/09/22/3","http://www.openwall.com/lists/oss-security/2023/09/22/4","http://www.openwall.com/lists/oss-security/2023/09/22/5","http://www.openwall.com/lists/oss-security/2023/09/22/6","http://www.openwall.com/lists/oss-security/2023/09/22/7","http://www.openwall.com/lists/oss-security/2023/09/22/8","http://www.openwall.com/lists/oss-security/2023/09/26/1","http://www.openwall.com/lists/oss-security/2023/09/26/7","http://www.openwall.com/lists/oss-security/2023/09/28/1","http://www.openwall.com/lists/oss-security/2023/09/28/2","http://www.openwall.com/lists/oss-security/2023/09/28/4","https://adamcaudill.com/2023/09/14/whose-cve-is-it-anyway/","https://blog.isosceles.com/the-webp-0day/","https://bugzilla.suse.com/show_bug.cgi?id=1215231","https://chromereleases.googleblog.com/2023/09/stable-channel-update-for-desktop_11.html","https://crbug.com/1479274","https://en.bandisoft.com/honeyview/history/","https://github.com/webmproject/libwebp/commit/902bc9190331343b2017211debcec8d2ab87e17a","https://github.com/webmproject/libwebp/releases/tag/v1.3.2","https://lists.debian.org/debian-lts-announce/2023/09/msg00015.html","https://lists.debian.org/debian-lts-announce/2023/09/msg00016.html","https://lists.debian.org/debian-lts-announce/2023/09/msg00017.html","https://lists.fedoraproject.org/archives/list/package-announce@lists.fedoraproject.org/message/6T655QF7CQ3DYAMPFV7IECQYGDEUIVVT/","https://lists.fedoraproject.org/archives/list/package-announce@lists.fedoraproject.org/message/FYYKLG6CRGEDTNRBSU26EEWAO6D6U645/","https://lists.fedoraproject.org/archives/list/package-announce@lists.fedoraproject.org/message/KUQ7CTX3W372X3UY56VVNAHCH6H2F4X3/","https://lists.fedoraproject.org/archives/list/package-announce@lists.fedoraproject.org/message/OZDGWWMJREPAGKWCJKSCM4WYLANSKIFX/","https://lists.fedoraproject.org/archives/list/package-announce@lists.fedoraproject.org/message/PYZV7TMKF4QHZ54SFJX54BDN52VHGGCX/","https://lists.fedoraproject.org/archives/list/package-announce@lists.fedoraproject.org/message/WHOLML7N2G5KCAZXFWC5IDFFHSQS5SDB/","https://lists.fedoraproject.org/archives/list/package-announce@lists.fedoraproject.org/message/WTRUIS3564P7ZLM2S2IH4Y4KZ327LI4I/","https://msrc.microsoft.com/update-guide/vulnerability/CVE-2023-4863","https://news.ycombinator.com/item?id=37478403","https://security-tracker.debian.org/tracker/CVE-2023-4863","https://security.gentoo.org/glsa/202309-05","https://security.gentoo.org/glsa/202401-10","https://security.netapp.com/advisory/ntap-20230929-0011/","https://sethmlarson.dev/security-developer-in-residence-weekly-report-16","https://stackdiary.com/critical-vulnerability-in-webp-codec-cve-2023-4863/","https://www.bentley.com/advisories/be-2023-0001/","https://www.bleepingcomputer.com/news/google/google-fixes-another-chrome-zero-day-bug-exploited-in-attacks/","https://www.debian.org/security/2023/dsa-5496","https://www.debian.org/security/2023/dsa-5497","https://www.debian.org/security/2023/dsa-5498","https://www.mozilla.org/en-US/security/advisories/mfsa2023-40/","http://www.openwall.com/lists/oss-security/2023/09/21/4","http://www.openwall.com/lists/oss-security/2023/09/22/1","http://www.openwall.com/lists/oss-security/2023/09/22/3","http://www.openwall.com/lists/oss-security/2023/09/22/4","http://www.openwall.com/lists/oss-security/2023/09/22/5","http://www.openwall.com/lists/oss-security/2023/09/22/6","http://www.openwall.com/lists/oss-security/2023/09/22/7","http://www.openwall.com/lists/oss-security/2023/09/22/8","http://www.openwall.com/lists/oss-security/2023/09/26/1","http://www.openwall.com/lists/oss-security/2023/09/26/7","http://www.openwall.com/lists/oss-security/2023/09/28/1","http://www.openwall.com/lists/oss-security/2023/09/28/2","http://www.openwall.com/lists/oss-security/2023/09/28/4","https://adamcaudill.com/2023/09/14/whose-cve-is-it-anyway/","https://blog.isosceles.com/the-webp-0day/","https://bugzilla.suse.com/show_bug.cgi?id=1215231","https://chromereleases.googleblog.com/2023/09/stable-channel-update-for-desktop_11.html","https://crbug.com/1479274","https://en.bandisoft.com/honeyview/history/","https://github.com/webmproject/libwebp/commit/902bc9190331343b2017211debcec8d2ab87e17a","https://github.com/webmproject/libwebp/releases/tag/v1.3.2","https://lists.debian.org/debian-lts-announce/2023/09/msg00015.html","https://lists.debian.org/debian-lts-announce/2023/09/msg00016.html","https://lists.debian.org/debian-lts-announce/2023/09/msg00017.html","https://lists.fedoraproject.org/archives/list/package-announce@lists.fedoraproject.org/message/6T655QF7CQ3DYAMPFV7IECQYGDEUIVVT/","https://lists.fedoraproject.org/archives/list/package-announce@lists.fedoraproject.org/message/FYYKLG6CRGEDTNRBSU26EEWAO6D6U645/","https://lists.fedoraproject.org/archives/list/package-announce@lists.fedoraproject.org/message/KUQ7CTX3W372X3UY56VVNAHCH6H2F4X3/","https://lists.fedoraproject.org/archives/list/package-announce@lists.fedoraproject.org/message/OZDGWWMJREPAGKWCJKSCM4WYLANSKIFX/","https://lists.fedoraproject.org/archives/list/package-announce@lists.fedoraproject.org/message/PYZV7TMKF4QHZ54SFJX54BDN52VHGGCX/","https://lists.fedoraproject.org/archives/list/package-announce@lists.fedoraproject.org/message/WHOLML7N2G5KCAZXFWC5IDFFHSQS5SDB/","https://lists.fedoraproject.org/archives/list/package-announce@lists.fedoraproject.org/message/WTRUIS3564P7ZLM2S2IH4Y4KZ327LI4I/","https://msrc.microsoft.com/update-guide/vulnerability/CVE-2023-4863","https://news.ycombinator.com/item?id=37478403","https://security-tracker.debian.org/tracker/CVE-2023-4863","https://security.gentoo.org/glsa/202309-05","https://security.gentoo.org/glsa/202401-10","https://security.netapp.com/advisory/ntap-20230929-0011/","https://sethmlarson.dev/security-developer-in-residence-weekly-report-16","https://stackdiary.com/critical-vulnerability-in-webp-codec-cve-2023-4863/","https://www.bentley.com/advisories/be-2023-0001/","https://www.bleepingcomputer.com/news/google/google-fixes-another-chrome-zero-day-bug-exploited-in-attacks/","https://www.debian.org/security/2023/dsa-5496","https://www.debian.org/security/2023/dsa-5497","https://www.debian.org/security/2023/dsa-5498","https://www.mozilla.org/en-US/security/advisories/mfsa2023-40/","https://www.vicarius.io/vsociety/posts/zero-day-webp-vulnerability-cve-2023-4863"],"description":"Heap buffer overflow in libwebp in Google Chrome prior to 116.0.5845.187 and libwebp 1.3.2 allowed a remote attacker to perform an out of bounds memory write via a crafted HTML page. (Chromium security severity: Critical)","cvss":[{"source":"nvd@nist.gov","type":"Primary","version":"3.1","vector":"CVSS:3.1/AV:N/AC:L/PR:N/UI:R/S:U/C:H/I:H/A:H","metrics":{"baseScore":8.8,"exploitabilityScore":2.9,"impactScore":5.9},"vendorMetadata":{}},{"source":"134c704f-9b21-4f2e-91b3-4a467353bcc0","type":"Secondary","version":"3.1","vector":"CVSS:3.1/AV:N/AC:L/PR:N/UI:R/S:U/C:H/I:H/A:H","metrics":{"baseScore":8.8,"exploitabilityScore":2.9,"impactScore":5.9},"vendorMetadata":{}}],"knownExploited":[{"cve":"CVE-2023-4863","vendorProject":"Google","product":"Chromium WebP","dateAdded":"2023-09-13","requiredAction":"Apply mitigations per vendor instructions or discontinue use of the product if mitigations are unavailable.","dueDate":"2023-10-04","knownRansomwareCampaignUse":"unknown","urls":["https://chromereleases.googleblog.com/2023/09/stable-channel-update-for-desktop_11.html?m=1","https://nvd.nist.gov/vuln/detail/CVE-2023-4863"],"cwes":["CWE-787"]}],"epss":[{"cve":"CVE-2023-4863","epss":0.93991,"percentile":0.9988,"date":"2025-06-02"}]}],"matchDetails":[{"type":"exact-direct-match","matcher":"python-matcher","searchedBy":{"language":"python","namespace":"github:language:python","package":{"name":"pillow","version":"9.0.1"}},"found":{"versionConstraint":"<10.0.1 (python)","vulnerabilityID":"GHSA-j7hp-h8jx-5ppr"},"fix":{"suggestedVersion":"10.0.1"}}],"artifact":{"id":"a191cb575c535611","name":"pillow","version":"9.0.1","type":"python","locations":[{"path":"/usr/lib/python3/dist-packages/Pillow-9.0.1.egg-info/PKG-INFO","accessPath":"/usr/lib/python3/dist-packages/Pillow-9.0.1.egg-info/PKG-INFO","annotations":{"evidence":"primary"}},{"path":"/usr/lib/python3/dist-packages/Pillow-9.0.1.egg-info/top_level.txt","accessPath":"/usr/lib/python3/dist-packages/Pillow-9.0.1.egg-info/top_level.txt","annotations":{"evidence":"supporting"}}],"language":"python","licenses":["HPND"],"cpes":["cpe:2.3:a:alex_clark_(pil_fork_author)_project:python-pillow:9.0.1:*:*:*:*:*:*:*","cpe:2.3:a:alex_clark_(pil_fork_author)_project:python_pillow:9.0.1:*:*:*:*:*:*:*","cpe:2.3:a:alex_clark_(pil_fork_author)project:python-pillow:9.0.1:*:*:*:*:*:*:*","cpe:2.3:a:alex_clark_(pil_fork_author)project:python_pillow:9.0.1:*:*:*:*:*:*:*","cpe:2.3:a:alex_clark_(pil_fork_author)_project:pillow:9.0.1:*:*:*:*:*:*:*","cpe:2.3:a:alex_clark_(pil_fork_author):python-pillow:9.0.1:*:*:*:*:*:*:*","cpe:2.3:a:alex_clark_(pil_fork_author):python_pillow:9.0.1:*:*:*:*:*:*:*","cpe:2.3:a:alex_clark_(pil_fork_author)project:pillow:9.0.1:*:*:*:*:*:*:*","cpe:2.3:a:alex_clark_(pil_fork_author):pillow:9.0.1:*:*:*:*:*:*:*","cpe:2.3:a:aclark_project:python-pillow:9.0.1:*:*:*:*:*:*:*","cpe:2.3:a:aclark_project:python_pillow:9.0.1:*:*:*:*:*:*:*","cpe:2.3:a:aclarkproject:python-pillow:9.0.1:*:*:*:*:*:*:*","cpe:2.3:a:aclarkproject:python_pillow:9.0.1:*:*:*:*:*:*:*","cpe:2.3:a:python-pillow:python-pillow:9.0.1:*:*:*:*:*:*:*","cpe:2.3:a:python-pillow:python_pillow:9.0.1:*:*:*:*:*:*:*","cpe:2.3:a:python_pillow:python-pillow:9.0.1:*:*:*:*:*:*:*","cpe:2.3:a:python_pillow:python_pillow:9.0.1:*:*:*:*:*:*:*","cpe:2.3:a:aclark_project:pillow:9.0.1:*:*:*:*:*:*:*","cpe:2.3:a:aclark:python-pillow:9.0.1:*:*:*:*:*:*:*","cpe:2.3:a:aclark:python_pillow:9.0.1:*:*:*:*:*:*:*","cpe:2.3:a:aclarkproject:pillow:9.0.1:*:*:*:*:*:*:*","cpe:2.3:a:pillow:python-pillow:9.0.1:*:*:*:*:*:*:*","cpe:2.3:a:pillow:python_pillow:9.0.1:*:*:*:*:*:*:*","cpe:2.3:a:python-pillow:pillow:9.0.1:*:*:*:*:*:*:*","cpe:2.3:a:python:python-pillow:9.0.1:*:*:*:*:*:*:*","cpe:2.3:a:python:python_pillow:9.0.1:*:*:*:*:*:*:*","cpe:2.3:a:python_pillow:pillow:9.0.1:*:*:*:*:*:*:*","cpe:2.3:a:aclark:pillow:9.0.1:*:*:*:*:*:*:*","cpe:2.3:a:pillow:pillow:9.0.1:*:*:*:*:*:*:*","cpe:2.3:a:python:pillow:9.0.1:*:*:*:*:*:*:*"],"purl":"pkg:pypi/pillow@9.0.1","upstreams":[]}},{"vulnerability":{"id":"GHSA-j8r2-6x86-q33q","dataSource":"https://github.com/advisories/GHSA-j8r2-6x86-q33q","namespace":"github:language:python","severity":"Medium","urls":[],"description":"Unintended leak of Proxy-Authorization header in requests","cvss":[{"type":"Secondary","version":"3.1","vector":"CVSS:3.1/AV:N/AC:H/PR:N/UI:R/S:C/C:H/I:N/A:N","metrics":{"baseScore":6.1,"exploitabilityScore":1.7,"impactScore":4},"vendorMetadata":{}}],"epss":[{"cve":"CVE-2023-32681","epss":0.04731,"percentile":0.88915,"date":"2025-06-02"}],"fix":{"versions":["2.31.0"],"state":"fixed"},"advisories":[],"risk":2.6257049999999995},"relatedVulnerabilities":[{"id":"CVE-2023-32681","dataSource":"https://nvd.nist.gov/vuln/detail/CVE-2023-32681","namespace":"nvd:cpe","severity":"Medium","urls":["https://github.com/psf/requests/commit/74ea7cf7a6a27a4eeb2ae24e162bcc942a6706d5","https://github.com/psf/requests/releases/tag/v2.31.0","https://github.com/psf/requests/security/advisories/GHSA-j8r2-6x86-q33q","https://lists.debian.org/debian-lts-announce/2023/06/msg00018.html","https://lists.fedoraproject.org/archives/list/package-announce@lists.fedoraproject.org/message/AW7HNFGYP44RT3DUDQXG2QT3OEV2PJ7Y/","https://lists.fedoraproject.org/archives/list/package-announce@lists.fedoraproject.org/message/KOYASTZDGQG2BWLSNBPL3TQRL2G7QYNZ/","https://security.gentoo.org/glsa/202309-08","https://github.com/psf/requests/commit/74ea7cf7a6a27a4eeb2ae24e162bcc942a6706d5","https://github.com/psf/requests/releases/tag/v2.31.0","https://github.com/psf/requests/security/advisories/GHSA-j8r2-6x86-q33q","https://lists.debian.org/debian-lts-announce/2023/06/msg00018.html","https://lists.fedoraproject.org/archives/list/package-announce@lists.fedoraproject.org/message/AW7HNFGYP44RT3DUDQXG2QT3OEV2PJ7Y/","https://lists.fedoraproject.org/archives/list/package-announce@lists.fedoraproject.org/message/KOYASTZDGQG2BWLSNBPL3TQRL2G7QYNZ/","https://security.gentoo.org/glsa/202309-08"],"description":"Requests is a HTTP library. Since Requests 2.3.0, Requests has been leaking Proxy-Authorization headers to destination servers when redirected to an HTTPS endpoint. This is a product of how we use `rebuild_proxies` to reattach the `Proxy-Authorization` header to requests. For HTTP connections sent through the tunnel, the proxy will identify the header in the request itself and remove it prior to forwarding to the destination server. However when sent over HTTPS, the `Proxy-Authorization` header must be sent in the CONNECT request as the proxy has no visibility into the tunneled request. This results in Requests forwarding proxy credentials to the destination server unintentionally, allowing a malicious actor to potentially exfiltrate sensitive information. This issue has been patched in version 2.31.0.","cvss":[{"source":"nvd@nist.gov","type":"Primary","version":"3.1","vector":"CVSS:3.1/AV:N/AC:H/PR:N/UI:R/S:C/C:H/I:N/A:N","metrics":{"baseScore":6.1,"exploitabilityScore":1.7,"impactScore":4},"vendorMetadata":{}},{"source":"security-advisories@github.com","type":"Secondary","version":"3.1","vector":"CVSS:3.1/AV:N/AC:H/PR:N/UI:R/S:C/C:H/I:N/A:N","metrics":{"baseScore":6.1,"exploitabilityScore":1.7,"impactScore":4},"vendorMetadata":{}}],"epss":[{"cve":"CVE-2023-32681","epss":0.04731,"percentile":0.88915,"date":"2025-06-02"}]}],"matchDetails":[{"type":"exact-direct-match","matcher":"python-matcher","searchedBy":{"language":"python","namespace":"github:language:python","package":{"name":"requests","version":"2.25.1"}},"found":{"versionConstraint":">=2.3.0,<2.31.0 (python)","vulnerabilityID":"GHSA-j8r2-6x86-q33q"},"fix":{"suggestedVersion":"2.31.0"}}],"artifact":{"id":"52e214e010c57a74","name":"requests","version":"2.25.1","type":"python","locations":[{"path":"/usr/lib/python3/dist-packages/requests-2.25.1.egg-info/PKG-INFO","accessPath":"/usr/lib/python3/dist-packages/requests-2.25.1.egg-info/PKG-INFO","annotations":{"evidence":"primary"}},{"path":"/usr/lib/python3/dist-packages/requests-2.25.1.egg-info/top_level.txt","accessPath":"/usr/lib/python3/dist-packages/requests-2.25.1.egg-info/top_level.txt","annotations":{"evidence":"supporting"}}],"language":"python","licenses":["Apache 2.0"],"cpes":["cpe:2.3:a:python:requests:2.25.1:*:*:*:*:*:*:*"],"purl":"pkg:pypi/requests@2.25.1","upstreams":[]}},{"vulnerability":{"id":"GHSA-v845-jxx5-vc9f","dataSource":"https://github.com/advisories/GHSA-v845-jxx5-vc9f","namespace":"github:language:python","severity":"High","urls":[],"description":"`Cookie` HTTP header isn't stripped on cross-origin redirects","cvss":[{"type":"Secondary","version":"3.1","vector":"CVSS:3.1/AV:N/AC:H/PR:H/UI:N/S:U/C:H/I:H/A:N","metrics":{"baseScore":5.9,"exploitabilityScore":0.8,"impactScore":5.2},"vendorMetadata":{}}],"epss":[{"cve":"CVE-2023-43804","epss":0.00554,"percentile":0.67074,"date":"2025-06-02"}],"fix":{"versions":["1.26.17"],"state":"fixed"},"advisories":[],"risk":0.37118},"relatedVulnerabilities":[{"id":"CVE-2023-43804","dataSource":"https://nvd.nist.gov/vuln/detail/CVE-2023-43804","namespace":"nvd:cpe","severity":"High","urls":["https://github.com/urllib3/urllib3/commit/01220354d389cd05474713f8c982d05c9b17aafb","https://github.com/urllib3/urllib3/commit/644124ecd0b6e417c527191f866daa05a5a2056d","https://github.com/urllib3/urllib3/security/advisories/GHSA-v845-jxx5-vc9f","https://lists.debian.org/debian-lts-announce/2023/10/msg00012.html","https://lists.fedoraproject.org/archives/list/package-announce@lists.fedoraproject.org/message/5F5CUBAN5XMEBVBZPHFITBLMJV5FIJJ5/","https://lists.fedoraproject.org/archives/list/package-announce@lists.fedoraproject.org/message/I3PR7C6RJ6JUBQKIJ644DMIJSUP36VDY/","https://lists.fedoraproject.org/archives/list/package-announce@lists.fedoraproject.org/message/NDAGZXYJ7H2G3SB47M453VQVNAWKAEJJ/","https://github.com/urllib3/urllib3/commit/01220354d389cd05474713f8c982d05c9b17aafb","https://github.com/urllib3/urllib3/commit/644124ecd0b6e417c527191f866daa05a5a2056d","https://github.com/urllib3/urllib3/security/advisories/GHSA-v845-jxx5-vc9f","https://lists.debian.org/debian-lts-announce/2023/10/msg00012.html","https://lists.fedoraproject.org/archives/list/package-announce@lists.fedoraproject.org/message/5F5CUBAN5XMEBVBZPHFITBLMJV5FIJJ5/","https://lists.fedoraproject.org/archives/list/package-announce@lists.fedoraproject.org/message/I3PR7C6RJ6JUBQKIJ644DMIJSUP36VDY/","https://lists.fedoraproject.org/archives/list/package-announce@lists.fedoraproject.org/message/NDAGZXYJ7H2G3SB47M453VQVNAWKAEJJ/","https://security.netapp.com/advisory/ntap-20241213-0007/","https://www.vicarius.io/vsociety/posts/cve-2023-43804-urllib3-vulnerability-3"],"description":"urllib3 is a user-friendly HTTP client library for Python. urllib3 doesn't treat the `Cookie` HTTP header special or provide any helpers for managing cookies over HTTP, that is the responsibility of the user. However, it is possible for a user to specify a `Cookie` header and unknowingly leak information via HTTP redirects to a different origin if that user doesn't disable redirects explicitly. This issue has been patched in urllib3 version 1.26.17 or 2.0.5.","cvss":[{"source":"nvd@nist.gov","type":"Primary","version":"3.1","vector":"CVSS:3.1/AV:N/AC:L/PR:L/UI:N/S:U/C:H/I:H/A:N","metrics":{"baseScore":8.1,"exploitabilityScore":2.9,"impactScore":5.2},"vendorMetadata":{}},{"source":"security-advisories@github.com","type":"Secondary","version":"3.1","vector":"CVSS:3.1/AV:N/AC:H/PR:H/UI:N/S:U/C:H/I:H/A:N","metrics":{"baseScore":5.9,"exploitabilityScore":0.8,"impactScore":5.2},"vendorMetadata":{}}],"epss":[{"cve":"CVE-2023-43804","epss":0.00554,"percentile":0.67074,"date":"2025-06-02"}]}],"matchDetails":[{"type":"exact-direct-match","matcher":"python-matcher","searchedBy":{"language":"python","namespace":"github:language:python","package":{"name":"urllib3","version":"1.26.5"}},"found":{"versionConstraint":"<1.26.17 (python)","vulnerabilityID":"GHSA-v845-jxx5-vc9f"},"fix":{"suggestedVersion":"1.26.17"}}],"artifact":{"id":"043d634961733646","name":"urllib3","version":"1.26.5","type":"python","locations":[{"path":"/usr/lib/python3/dist-packages/urllib3-1.26.5.egg-info/PKG-INFO","accessPath":"/usr/lib/python3/dist-packages/urllib3-1.26.5.egg-info/PKG-INFO","annotations":{"evidence":"primary"}},{"path":"/usr/lib/python3/dist-packages/urllib3-1.26.5.egg-info/top_level.txt","accessPath":"/usr/lib/python3/dist-packages/urllib3-1.26.5.egg-info/top_level.txt","annotations":{"evidence":"supporting"}}],"language":"python","licenses":["MIT"],"cpes":["cpe:2.3:a:python:urllib3:1.26.5:*:*:*:*:*:*:*"],"purl":"pkg:pypi/urllib3@1.26.5","upstreams":[]}},{"vulnerability":{"id":"GHSA-3f63-hfp8-52jq","dataSource":"https://github.com/advisories/GHSA-3f63-hfp8-52jq","namespace":"github:language:python","severity":"Critical","urls":[],"description":"Arbitrary Code Execution in Pillow","cvss":[{"type":"Secondary","version":"3.1","vector":"CVSS:3.1/AV:N/AC:H/PR:N/UI:N/S:U/C:H/I:H/A:H","metrics":{"baseScore":8.1,"exploitabilityScore":2.3,"impactScore":5.9},"vendorMetadata":{}}],"epss":[{"cve":"CVE-2023-50447","epss":0.00408,"percentile":0.6039,"date":"2025-06-02"}],"fix":{"versions":["10.2.0"],"state":"fixed"},"advisories":[],"risk":0.34884},"relatedVulnerabilities":[{"id":"CVE-2023-50447","dataSource":"https://nvd.nist.gov/vuln/detail/CVE-2023-50447","namespace":"nvd:cpe","severity":"High","urls":["http://www.openwall.com/lists/oss-security/2024/01/20/1","https://devhub.checkmarx.com/cve-details/CVE-2023-50447/","https://duartecsantos.github.io/2024-01-02-CVE-2023-50447/","https://github.com/python-pillow/Pillow/releases","https://lists.debian.org/debian-lts-announce/2024/01/msg00019.html","http://www.openwall.com/lists/oss-security/2024/01/20/1","https://devhub.checkmarx.com/cve-details/CVE-2023-50447/","https://duartecsantos.github.io/2024-01-02-CVE-2023-50447/","https://github.com/python-pillow/Pillow/releases","https://lists.debian.org/debian-lts-announce/2024/01/msg00019.html"],"description":"Pillow through 10.1.0 allows PIL.ImageMath.eval Arbitrary Code Execution via the environment parameter, a different vulnerability than CVE-2022-22817 (which was about the expression parameter).","cvss":[{"source":"nvd@nist.gov","type":"Primary","version":"3.1","vector":"CVSS:3.1/AV:N/AC:H/PR:N/UI:N/S:U/C:H/I:H/A:H","metrics":{"baseScore":8.1,"exploitabilityScore":2.3,"impactScore":5.9},"vendorMetadata":{}},{"source":"134c704f-9b21-4f2e-91b3-4a467353bcc0","type":"Secondary","version":"3.1","vector":"CVSS:3.1/AV:N/AC:H/PR:N/UI:N/S:U/C:H/I:H/A:H","metrics":{"baseScore":8.1,"exploitabilityScore":2.3,"impactScore":5.9},"vendorMetadata":{}}],"epss":[{"cve":"CVE-2023-50447","epss":0.00408,"percentile":0.6039,"date":"2025-06-02"}]}],"matchDetails":[{"type":"exact-direct-match","matcher":"python-matcher","searchedBy":{"language":"python","namespace":"github:language:python","package":{"name":"pillow","version":"9.0.1"}},"found":{"versionConstraint":"<10.2.0 (python)","vulnerabilityID":"GHSA-3f63-hfp8-52jq"},"fix":{"suggestedVersion":"10.2.0"}}],"artifact":{"id":"a191cb575c535611","name":"pillow","version":"9.0.1","type":"python","locations":[{"path":"/usr/lib/python3/dist-packages/Pillow-9.0.1.egg-info/PKG-INFO","accessPath":"/usr/lib/python3/dist-packages/Pillow-9.0.1.egg-info/PKG-INFO","annotations":{"evidence":"primary"}},{"path":"/usr/lib/python3/dist-packages/Pillow-9.0.1.egg-info/top_level.txt","accessPath":"/usr/lib/python3/dist-packages/Pillow-9.0.1.egg-info/top_level.txt","annotations":{"evidence":"supporting"}}],"language":"python","licenses":["HPND"],"cpes":["cpe:2.3:a:alex_clark_(pil_fork_author)_project:python-pillow:9.0.1:*:*:*:*:*:*:*","cpe:2.3:a:alex_clark_(pil_fork_author)_project:python_pillow:9.0.1:*:*:*:*:*:*:*","cpe:2.3:a:alex_clark_(pil_fork_author)project:python-pillow:9.0.1:*:*:*:*:*:*:*","cpe:2.3:a:alex_clark_(pil_fork_author)project:python_pillow:9.0.1:*:*:*:*:*:*:*","cpe:2.3:a:alex_clark_(pil_fork_author)_project:pillow:9.0.1:*:*:*:*:*:*:*","cpe:2.3:a:alex_clark_(pil_fork_author):python-pillow:9.0.1:*:*:*:*:*:*:*","cpe:2.3:a:alex_clark_(pil_fork_author):python_pillow:9.0.1:*:*:*:*:*:*:*","cpe:2.3:a:alex_clark_(pil_fork_author)project:pillow:9.0.1:*:*:*:*:*:*:*","cpe:2.3:a:alex_clark_(pil_fork_author):pillow:9.0.1:*:*:*:*:*:*:*","cpe:2.3:a:aclark_project:python-pillow:9.0.1:*:*:*:*:*:*:*","cpe:2.3:a:aclark_project:python_pillow:9.0.1:*:*:*:*:*:*:*","cpe:2.3:a:aclarkproject:python-pillow:9.0.1:*:*:*:*:*:*:*","cpe:2.3:a:aclarkproject:python_pillow:9.0.1:*:*:*:*:*:*:*","cpe:2.3:a:python-pillow:python-pillow:9.0.1:*:*:*:*:*:*:*","cpe:2.3:a:python-pillow:python_pillow:9.0.1:*:*:*:*:*:*:*","cpe:2.3:a:python_pillow:python-pillow:9.0.1:*:*:*:*:*:*:*","cpe:2.3:a:python_pillow:python_pillow:9.0.1:*:*:*:*:*:*:*","cpe:2.3:a:aclark_project:pillow:9.0.1:*:*:*:*:*:*:*","cpe:2.3:a:aclark:python-pillow:9.0.1:*:*:*:*:*:*:*","cpe:2.3:a:aclark:python_pillow:9.0.1:*:*:*:*:*:*:*","cpe:2.3:a:aclarkproject:pillow:9.0.1:*:*:*:*:*:*:*","cpe:2.3:a:pillow:python-pillow:9.0.1:*:*:*:*:*:*:*","cpe:2.3:a:pillow:python_pillow:9.0.1:*:*:*:*:*:*:*","cpe:2.3:a:python-pillow:pillow:9.0.1:*:*:*:*:*:*:*","cpe:2.3:a:python:python-pillow:9.0.1:*:*:*:*:*:*:*","cpe:2.3:a:python:python_pillow:9.0.1:*:*:*:*:*:*:*","cpe:2.3:a:python_pillow:pillow:9.0.1:*:*:*:*:*:*:*","cpe:2.3:a:aclark:pillow:9.0.1:*:*:*:*:*:*:*","cpe:2.3:a:pillow:pillow:9.0.1:*:*:*:*:*:*:*","cpe:2.3:a:python:pillow:9.0.1:*:*:*:*:*:*:*"],"purl":"pkg:pypi/pillow@9.0.1","upstreams":[]}},{"vulnerability":{"id":"GHSA-5rjg-fvgr-3xxf","dataSource":"https://github.com/advisories/GHSA-5rjg-fvgr-3xxf","namespace":"github:language:python","severity":"High","urls":[],"description":"setuptools has a path traversal vulnerability in PackageIndex.download that leads to Arbitrary File Write","cvss":[],"epss":[{"cve":"CVE-2025-47273","epss":0.00435,"percentile":0.6206,"date":"2025-06-02"}],"fix":{"versions":["78.1.1"],"state":"fixed"},"advisories":[],"risk":0.32625},"relatedVulnerabilities":[{"id":"CVE-2025-47273","dataSource":"https://nvd.nist.gov/vuln/detail/CVE-2025-47273","namespace":"nvd:cpe","severity":"High","urls":["https://github.com/pypa/setuptools/blob/6ead555c5fb29bc57fe6105b1bffc163f56fd558/setuptools/package_index.py#L810C1-L825C88","https://github.com/pypa/setuptools/commit/250a6d17978f9f6ac3ac887091f2d32886fbbb0b","https://github.com/pypa/setuptools/issues/4946","https://github.com/pypa/setuptools/security/advisories/GHSA-5rjg-fvgr-3xxf","https://lists.debian.org/debian-lts-announce/2025/05/msg00035.html","https://github.com/pypa/setuptools/issues/4946"],"description":"setuptools is a package that allows users to download, build, install, upgrade, and uninstall Python packages. A path traversal vulnerability in `PackageIndex` is present in setuptools prior to version 78.1.1. An attacker would be allowed to write files to arbitrary locations on the filesystem with the permissions of the process running the Python code, which could escalate to remote code execution depending on the context. Version 78.1.1 fixes the issue.","cvss":[{"source":"security-advisories@github.com","type":"Secondary","version":"4.0","vector":"CVSS:4.0/AV:N/AC:L/AT:N/PR:N/UI:N/VC:N/VI:H/VA:N/SC:N/SI:N/SA:N/E:P/CR:X/IR:X/AR:X/MAV:X/MAC:X/MAT:X/MPR:X/MUI:X/MVC:X/MVI:X/MVA:X/MSC:X/MSI:X/MSA:X/S:X/AU:X/R:X/V:X/RE:X/U:X","metrics":{"baseScore":7.7},"vendorMetadata":{}}],"epss":[{"cve":"CVE-2025-47273","epss":0.00435,"percentile":0.6206,"date":"2025-06-02"}]}],"matchDetails":[{"type":"exact-direct-match","matcher":"python-matcher","searchedBy":{"language":"python","namespace":"github:language:python","package":{"name":"setuptools","version":"75.6.0"}},"found":{"versionConstraint":"<78.1.1 (python)","vulnerabilityID":"GHSA-5rjg-fvgr-3xxf"},"fix":{"suggestedVersion":"78.1.1"}}],"artifact":{"id":"1ce1e218b484c3b7","name":"setuptools","version":"75.6.0","type":"python","locations":[{"path":"/lib/python3.10/site-packages/setuptools-75.6.0.dist-info/METADATA","accessPath":"/lib/python3.10/site-packages/setuptools-75.6.0.dist-info/METADATA","annotations":{"evidence":"primary"}},{"path":"/lib/python3.10/site-packages/setuptools-75.6.0.dist-info/RECORD","accessPath":"/lib/python3.10/site-packages/setuptools-75.6.0.dist-info/RECORD","annotations":{"evidence":"supporting"}},{"path":"/lib/python3.10/site-packages/setuptools-75.6.0.dist-info/top_level.txt","accessPath":"/lib/python3.10/site-packages/setuptools-75.6.0.dist-info/top_level.txt","annotations":{"evidence":"supporting"}}],"language":"python","licenses":["MIT"],"cpes":["cpe:2.3:a:python:setuptools:75.6.0:*:*:*:*:*:*:*"],"purl":"pkg:pypi/setuptools@75.6.0","upstreams":[]}},{"vulnerability":{"id":"GHSA-v3c5-jqr6-7qm8","dataSource":"https://github.com/advisories/GHSA-v3c5-jqr6-7qm8","namespace":"github:language:python","severity":"High","urls":[],"description":"Python Charmers Future denial of service vulnerability","cvss":[{"type":"Secondary","version":"3.1","vector":"CVSS:3.1/AV:N/AC:L/PR:N/UI:N/S:U/C:N/I:N/A:H","metrics":{"baseScore":7.5,"exploitabilityScore":3.9,"impactScore":3.6},"vendorMetadata":{}}],"epss":[{"cve":"CVE-2022-40899","epss":0.00419,"percentile":0.61099,"date":"2025-06-02"}],"fix":{"versions":["0.18.3"],"state":"fixed"},"advisories":[],"risk":0.31425000000000003},"relatedVulnerabilities":[{"id":"CVE-2022-40899","dataSource":"https://nvd.nist.gov/vuln/detail/CVE-2022-40899","namespace":"nvd:cpe","severity":"High","urls":["https://github.com/PythonCharmers/python-future/blob/master/src/future/backports/http/cookiejar.py#L215","https://github.com/PythonCharmers/python-future/pull/610","https://github.com/python/cpython/pull/17157","https://pypi.org/project/future/","https://pyup.io/posts/pyup-discovers-redos-vulnerabilities-in-top-python-packages/","https://github.com/PythonCharmers/python-future/blob/master/src/future/backports/http/cookiejar.py#L215","https://github.com/PythonCharmers/python-future/pull/610","https://github.com/python/cpython/pull/17157","https://pypi.org/project/future/","https://pyup.io/posts/pyup-discovers-redos-vulnerabilities-in-top-python-packages/"],"description":"An issue discovered in Python Charmers Future 0.18.2 and earlier allows remote attackers to cause a denial of service via crafted Set-Cookie header from malicious web server.","cvss":[{"source":"nvd@nist.gov","type":"Primary","version":"3.1","vector":"CVSS:3.1/AV:N/AC:L/PR:N/UI:N/S:U/C:N/I:N/A:H","metrics":{"baseScore":7.5,"exploitabilityScore":3.9,"impactScore":3.6},"vendorMetadata":{}},{"source":"134c704f-9b21-4f2e-91b3-4a467353bcc0","type":"Secondary","version":"3.1","vector":"CVSS:3.1/AV:N/AC:L/PR:N/UI:N/S:U/C:N/I:N/A:H","metrics":{"baseScore":7.5,"exploitabilityScore":3.9,"impactScore":3.6},"vendorMetadata":{}}],"epss":[{"cve":"CVE-2022-40899","epss":0.00419,"percentile":0.61099,"date":"2025-06-02"}]}],"matchDetails":[{"type":"exact-direct-match","matcher":"python-matcher","searchedBy":{"language":"python","namespace":"github:language:python","package":{"name":"future","version":"0.18.2"}},"found":{"versionConstraint":"<=0.18.2 (python)","vulnerabilityID":"GHSA-v3c5-jqr6-7qm8"},"fix":{"suggestedVersion":"0.18.3"}}],"artifact":{"id":"8ea8e68e8d7cab0c","name":"future","version":"0.18.2","type":"python","locations":[{"path":"/usr/lib/python3/dist-packages/future-0.18.2.egg-info/PKG-INFO","accessPath":"/usr/lib/python3/dist-packages/future-0.18.2.egg-info/PKG-INFO","annotations":{"evidence":"primary"}},{"path":"/usr/lib/python3/dist-packages/future-0.18.2.egg-info/top_level.txt","accessPath":"/usr/lib/python3/dist-packages/future-0.18.2.egg-info/top_level.txt","annotations":{"evidence":"supporting"}}],"language":"python","licenses":["MIT"],"cpes":["cpe:2.3:a:ed_schofield_project:python-future:0.18.2:*:*:*:*:*:*:*","cpe:2.3:a:ed_schofield_project:python_future:0.18.2:*:*:*:*:*:*:*","cpe:2.3:a:ed_schofieldproject:python-future:0.18.2:*:*:*:*:*:*:*","cpe:2.3:a:ed_schofieldproject:python_future:0.18.2:*:*:*:*:*:*:*","cpe:2.3:a:ed_schofield_project:future:0.18.2:*:*:*:*:*:*:*","cpe:2.3:a:python-future:python-future:0.18.2:*:*:*:*:*:*:*","cpe:2.3:a:python-future:python_future:0.18.2:*:*:*:*:*:*:*","cpe:2.3:a:python_future:python-future:0.18.2:*:*:*:*:*:*:*","cpe:2.3:a:python_future:python_future:0.18.2:*:*:*:*:*:*:*","cpe:2.3:a:ed_schofield:python-future:0.18.2:*:*:*:*:*:*:*","cpe:2.3:a:ed_schofield:python_future:0.18.2:*:*:*:*:*:*:*","cpe:2.3:a:ed_schofieldproject:future:0.18.2:*:*:*:*:*:*:*","cpe:2.3:a:ed_project:python-future:0.18.2:*:*:*:*:*:*:*","cpe:2.3:a:ed_project:python_future:0.18.2:*:*:*:*:*:*:*","cpe:2.3:a:edproject:python-future:0.18.2:*:*:*:*:*:*:*","cpe:2.3:a:edproject:python_future:0.18.2:*:*:*:*:*:*:*","cpe:2.3:a:future:python-future:0.18.2:*:*:*:*:*:*:*","cpe:2.3:a:future:python_future:0.18.2:*:*:*:*:*:*:*","cpe:2.3:a:python-future:future:0.18.2:*:*:*:*:*:*:*","cpe:2.3:a:python:python-future:0.18.2:*:*:*:*:*:*:*","cpe:2.3:a:python:python_future:0.18.2:*:*:*:*:*:*:*","cpe:2.3:a:python_future:future:0.18.2:*:*:*:*:*:*:*","cpe:2.3:a:ed_schofield:future:0.18.2:*:*:*:*:*:*:*","cpe:2.3:a:ed_project:future:0.18.2:*:*:*:*:*:*:*","cpe:2.3:a:ed:python-future:0.18.2:*:*:*:*:*:*:*","cpe:2.3:a:ed:python_future:0.18.2:*:*:*:*:*:*:*","cpe:2.3:a:edproject:future:0.18.2:*:*:*:*:*:*:*","cpe:2.3:a:future:future:0.18.2:*:*:*:*:*:*:*","cpe:2.3:a:python:future:0.18.2:*:*:*:*:*:*:*","cpe:2.3:a:ed:future:0.18.2:*:*:*:*:*:*:*"],"purl":"pkg:pypi/future@0.18.2","upstreams":[]}},{"vulnerability":{"id":"GHSA-jjg7-2v4v-x38h","dataSource":"https://github.com/advisories/GHSA-jjg7-2v4v-x38h","namespace":"github:language:python","severity":"Medium","urls":[],"description":"Internationalized Domain Names in Applications (IDNA) vulnerable to denial of service from specially crafted inputs to idna.encode","cvss":[{"type":"Secondary","version":"3.1","vector":"CVSS:3.1/AV:L/AC:L/PR:N/UI:N/S:U/C:N/I:N/A:H","metrics":{"baseScore":6.2,"exploitabilityScore":2.6,"impactScore":3.6},"vendorMetadata":{}}],"epss":[{"cve":"CVE-2024-3651","epss":0.00436,"percentile":0.62112,"date":"2025-06-02"}],"fix":{"versions":["3.7"],"state":"fixed"},"advisories":[],"risk":0.24416000000000004},"relatedVulnerabilities":[{"id":"CVE-2024-3651","dataSource":"https://nvd.nist.gov/vuln/detail/CVE-2024-3651","namespace":"nvd:cpe","severity":"High","urls":["https://github.com/kjd/idna/commit/1d365e17e10d72d0b7876316fc7b9ca0eebdd38d","https://huntr.com/bounties/93d78d07-d791-4b39-a845-cbfabc44aadb","https://github.com/kjd/idna/commit/1d365e17e10d72d0b7876316fc7b9ca0eebdd38d","https://huntr.com/bounties/93d78d07-d791-4b39-a845-cbfabc44aadb"],"description":"A vulnerability was identified in the kjd/idna library, specifically within the `idna.encode()` function, affecting version 3.6. The issue arises from the function's handling of crafted input strings, which can lead to quadratic complexity and consequently, a denial of service condition. This vulnerability is triggered by a crafted input that causes the `idna.encode()` function to process the input with considerable computational load, significantly increasing the processing time in a quadratic manner relative to the input size.","cvss":[{"source":"nvd@nist.gov","type":"Primary","version":"3.1","vector":"CVSS:3.1/AV:N/AC:L/PR:N/UI:N/S:U/C:N/I:N/A:H","metrics":{"baseScore":7.5,"exploitabilityScore":3.9,"impactScore":3.6},"vendorMetadata":{}},{"source":"security@huntr.dev","type":"Secondary","version":"3.0","vector":"CVSS:3.0/AV:L/AC:L/PR:N/UI:N/S:U/C:N/I:N/A:H","metrics":{"baseScore":6.2,"exploitabilityScore":2.6,"impactScore":3.6},"vendorMetadata":{}}],"epss":[{"cve":"CVE-2024-3651","epss":0.00436,"percentile":0.62112,"date":"2025-06-02"}]}],"matchDetails":[{"type":"exact-direct-match","matcher":"python-matcher","searchedBy":{"language":"python","namespace":"github:language:python","package":{"name":"idna","version":"3.3"}},"found":{"versionConstraint":"<3.7 (python)","vulnerabilityID":"GHSA-jjg7-2v4v-x38h"},"fix":{"suggestedVersion":"3.7"}}],"artifact":{"id":"30be7e210695bf55","name":"idna","version":"3.3","type":"python","locations":[{"path":"/usr/lib/python3/dist-packages/idna-3.3.egg-info/PKG-INFO","accessPath":"/usr/lib/python3/dist-packages/idna-3.3.egg-info/PKG-INFO","annotations":{"evidence":"primary"}},{"path":"/usr/lib/python3/dist-packages/idna-3.3.egg-info/top_level.txt","accessPath":"/usr/lib/python3/dist-packages/idna-3.3.egg-info/top_level.txt","annotations":{"evidence":"supporting"}}],"language":"python","licenses":["BSD-3-Clause"],"cpes":["cpe:2.3:a:kim_davies_project:python-idna:3.3:*:*:*:*:*:*:*","cpe:2.3:a:kim_davies_project:python_idna:3.3:*:*:*:*:*:*:*","cpe:2.3:a:kim_daviesproject:python-idna:3.3:*:*:*:*:*:*:*","cpe:2.3:a:kim_daviesproject:python_idna:3.3:*:*:*:*:*:*:*","cpe:2.3:a:kim_davies_project:idna:3.3:*:*:*:*:*:*:*","cpe:2.3:a:kim_project:python-idna:3.3:*:*:*:*:*:*:*","cpe:2.3:a:kim_project:python_idna:3.3:*:*:*:*:*:*:*","cpe:2.3:a:python-idna:python-idna:3.3:*:*:*:*:*:*:*","cpe:2.3:a:python-idna:python_idna:3.3:*:*:*:*:*:*:*","cpe:2.3:a:python_idna:python-idna:3.3:*:*:*:*:*:*:*","cpe:2.3:a:python_idna:python_idna:3.3:*:*:*:*:*:*:*","cpe:2.3:a:kim_davies:python-idna:3.3:*:*:*:*:*:*:*","cpe:2.3:a:kim_davies:python_idna:3.3:*:*:*:*:*:*:*","cpe:2.3:a:kim_daviesproject:idna:3.3:*:*:*:*:*:*:*","cpe:2.3:a:kimproject:python-idna:3.3:*:*:*:*:*:*:*","cpe:2.3:a:kimproject:python_idna:3.3:*:*:*:*:*:*:*","cpe:2.3:a:python:python-idna:3.3:*:*:*:*:*:*:*","cpe:2.3:a:python:python_idna:3.3:*:*:*:*:*:*:*","cpe:2.3:a:idna:python-idna:3.3:*:*:*:*:*:*:*","cpe:2.3:a:idna:python_idna:3.3:*:*:*:*:*:*:*","cpe:2.3:a:kim_project:idna:3.3:*:*:*:*:*:*:*","cpe:2.3:a:python-idna:idna:3.3:*:*:*:*:*:*:*","cpe:2.3:a:python_idna:idna:3.3:*:*:*:*:*:*:*","cpe:2.3:a:kim:python-idna:3.3:*:*:*:*:*:*:*","cpe:2.3:a:kim:python_idna:3.3:*:*:*:*:*:*:*","cpe:2.3:a:kim_davies:idna:3.3:*:*:*:*:*:*:*","cpe:2.3:a:kimproject:idna:3.3:*:*:*:*:*:*:*","cpe:2.3:a:python:idna:3.3:*:*:*:*:*:*:*","cpe:2.3:a:idna:idna:3.3:*:*:*:*:*:*:*","cpe:2.3:a:kim:idna:3.3:*:*:*:*:*:*:*"],"purl":"pkg:pypi/idna@3.3","upstreams":[]}},{"vulnerability":{"id":"GHSA-8ghj-p4vj-mr35","dataSource":"https://github.com/advisories/GHSA-8ghj-p4vj-mr35","namespace":"github:language:python","severity":"High","urls":[],"description":"Pillow Denial of Service vulnerability","cvss":[{"type":"Secondary","version":"3.1","vector":"CVSS:3.1/AV:N/AC:L/PR:N/UI:N/S:U/C:N/I:N/A:H","metrics":{"baseScore":7.5,"exploitabilityScore":3.9,"impactScore":3.6},"vendorMetadata":{}}],"epss":[{"cve":"CVE-2023-44271","epss":0.00148,"percentile":0.36591,"date":"2025-06-02"}],"fix":{"versions":["10.0.0"],"state":"fixed"},"advisories":[],"risk":0.11099999999999999},"relatedVulnerabilities":[{"id":"CVE-2023-44271","dataSource":"https://nvd.nist.gov/vuln/detail/CVE-2023-44271","namespace":"nvd:cpe","severity":"High","urls":["https://devhub.checkmarx.com/cve-details/CVE-2023-44271/","https://github.com/python-pillow/Pillow/commit/1fe1bb49c452b0318cad12ea9d97c3bef188e9a7","https://github.com/python-pillow/Pillow/pull/7244","https://lists.debian.org/debian-lts-announce/2024/03/msg00021.html","https://lists.fedoraproject.org/archives/list/package-announce%40lists.fedoraproject.org/message/N2JOEDUJDQLCUII2LQYZYSM7RJL2I3P4/","https://devhub.checkmarx.com/cve-details/CVE-2023-44271/","https://github.com/python-pillow/Pillow/commit/1fe1bb49c452b0318cad12ea9d97c3bef188e9a7","https://github.com/python-pillow/Pillow/pull/7244","https://lists.debian.org/debian-lts-announce/2024/03/msg00021.html","https://lists.fedoraproject.org/archives/list/package-announce%40lists.fedoraproject.org/message/N2JOEDUJDQLCUII2LQYZYSM7RJL2I3P4/"],"description":"An issue was discovered in Pillow before 10.0.0. It is a Denial of Service that uncontrollably allocates memory to process a given task, potentially causing a service to crash by having it run out of memory. This occurs for truetype in ImageFont when textlength in an ImageDraw instance operates on a long text argument.","cvss":[{"source":"nvd@nist.gov","type":"Primary","version":"3.1","vector":"CVSS:3.1/AV:N/AC:L/PR:N/UI:N/S:U/C:N/I:N/A:H","metrics":{"baseScore":7.5,"exploitabilityScore":3.9,"impactScore":3.6},"vendorMetadata":{}}],"epss":[{"cve":"CVE-2023-44271","epss":0.00148,"percentile":0.36591,"date":"2025-06-02"}]}],"matchDetails":[{"type":"exact-direct-match","matcher":"python-matcher","searchedBy":{"language":"python","namespace":"github:language:python","package":{"name":"pillow","version":"9.0.1"}},"found":{"versionConstraint":">=0,<10.0.0 (python)","vulnerabilityID":"GHSA-8ghj-p4vj-mr35"},"fix":{"suggestedVersion":"10.0.0"}}],"artifact":{"id":"a191cb575c535611","name":"pillow","version":"9.0.1","type":"python","locations":[{"path":"/usr/lib/python3/dist-packages/Pillow-9.0.1.egg-info/PKG-INFO","accessPath":"/usr/lib/python3/dist-packages/Pillow-9.0.1.egg-info/PKG-INFO","annotations":{"evidence":"primary"}},{"path":"/usr/lib/python3/dist-packages/Pillow-9.0.1.egg-info/top_level.txt","accessPath":"/usr/lib/python3/dist-packages/Pillow-9.0.1.egg-info/top_level.txt","annotations":{"evidence":"supporting"}}],"language":"python","licenses":["HPND"],"cpes":["cpe:2.3:a:alex_clark_(pil_fork_author)_project:python-pillow:9.0.1:*:*:*:*:*:*:*","cpe:2.3:a:alex_clark_(pil_fork_author)_project:python_pillow:9.0.1:*:*:*:*:*:*:*","cpe:2.3:a:alex_clark_(pil_fork_author)project:python-pillow:9.0.1:*:*:*:*:*:*:*","cpe:2.3:a:alex_clark_(pil_fork_author)project:python_pillow:9.0.1:*:*:*:*:*:*:*","cpe:2.3:a:alex_clark_(pil_fork_author)_project:pillow:9.0.1:*:*:*:*:*:*:*","cpe:2.3:a:alex_clark_(pil_fork_author):python-pillow:9.0.1:*:*:*:*:*:*:*","cpe:2.3:a:alex_clark_(pil_fork_author):python_pillow:9.0.1:*:*:*:*:*:*:*","cpe:2.3:a:alex_clark_(pil_fork_author)project:pillow:9.0.1:*:*:*:*:*:*:*","cpe:2.3:a:alex_clark_(pil_fork_author):pillow:9.0.1:*:*:*:*:*:*:*","cpe:2.3:a:aclark_project:python-pillow:9.0.1:*:*:*:*:*:*:*","cpe:2.3:a:aclark_project:python_pillow:9.0.1:*:*:*:*:*:*:*","cpe:2.3:a:aclarkproject:python-pillow:9.0.1:*:*:*:*:*:*:*","cpe:2.3:a:aclarkproject:python_pillow:9.0.1:*:*:*:*:*:*:*","cpe:2.3:a:python-pillow:python-pillow:9.0.1:*:*:*:*:*:*:*","cpe:2.3:a:python-pillow:python_pillow:9.0.1:*:*:*:*:*:*:*","cpe:2.3:a:python_pillow:python-pillow:9.0.1:*:*:*:*:*:*:*","cpe:2.3:a:python_pillow:python_pillow:9.0.1:*:*:*:*:*:*:*","cpe:2.3:a:aclark_project:pillow:9.0.1:*:*:*:*:*:*:*","cpe:2.3:a:aclark:python-pillow:9.0.1:*:*:*:*:*:*:*","cpe:2.3:a:aclark:python_pillow:9.0.1:*:*:*:*:*:*:*","cpe:2.3:a:aclarkproject:pillow:9.0.1:*:*:*:*:*:*:*","cpe:2.3:a:pillow:python-pillow:9.0.1:*:*:*:*:*:*:*","cpe:2.3:a:pillow:python_pillow:9.0.1:*:*:*:*:*:*:*","cpe:2.3:a:python-pillow:pillow:9.0.1:*:*:*:*:*:*:*","cpe:2.3:a:python:python-pillow:9.0.1:*:*:*:*:*:*:*","cpe:2.3:a:python:python_pillow:9.0.1:*:*:*:*:*:*:*","cpe:2.3:a:python_pillow:pillow:9.0.1:*:*:*:*:*:*:*","cpe:2.3:a:aclark:pillow:9.0.1:*:*:*:*:*:*:*","cpe:2.3:a:pillow:pillow:9.0.1:*:*:*:*:*:*:*","cpe:2.3:a:python:pillow:9.0.1:*:*:*:*:*:*:*"],"purl":"pkg:pypi/pillow@9.0.1","upstreams":[]}},{"vulnerability":{"id":"GHSA-m2vv-5vj5-2hm7","dataSource":"https://github.com/advisories/GHSA-m2vv-5vj5-2hm7","namespace":"github:language:python","severity":"High","urls":[],"description":"Pillow vulnerable to Data Amplification attack.","cvss":[{"type":"Secondary","version":"3.1","vector":"CVSS:3.1/AV:N/AC:L/PR:N/UI:N/S:U/C:N/I:N/A:H","metrics":{"baseScore":7.5,"exploitabilityScore":3.9,"impactScore":3.6},"vendorMetadata":{}}],"epss":[{"cve":"CVE-2022-45198","epss":0.00123,"percentile":0.32712,"date":"2025-06-02"}],"fix":{"versions":["9.2.0"],"state":"fixed"},"advisories":[],"risk":0.09225},"relatedVulnerabilities":[{"id":"CVE-2022-45198","dataSource":"https://nvd.nist.gov/vuln/detail/CVE-2022-45198","namespace":"nvd:cpe","severity":"High","urls":["https://bugs.gentoo.org/855683","https://cwe.mitre.org/data/definitions/409.html","https://github.com/python-pillow/Pillow/commit/11918eac0628ec8ac0812670d9838361ead2d6a4","https://github.com/python-pillow/Pillow/pull/6402","https://github.com/python-pillow/Pillow/releases/tag/9.2.0","https://security.gentoo.org/glsa/202211-10","https://bugs.gentoo.org/855683","https://cwe.mitre.org/data/definitions/409.html","https://github.com/python-pillow/Pillow/commit/11918eac0628ec8ac0812670d9838361ead2d6a4","https://github.com/python-pillow/Pillow/pull/6402","https://github.com/python-pillow/Pillow/releases/tag/9.2.0","https://security.gentoo.org/glsa/202211-10"],"description":"Pillow before 9.2.0 performs Improper Handling of Highly Compressed GIF Data (Data Amplification).","cvss":[{"source":"nvd@nist.gov","type":"Primary","version":"3.1","vector":"CVSS:3.1/AV:N/AC:L/PR:N/UI:N/S:U/C:N/I:N/A:H","metrics":{"baseScore":7.5,"exploitabilityScore":3.9,"impactScore":3.6},"vendorMetadata":{}}],"epss":[{"cve":"CVE-2022-45198","epss":0.00123,"percentile":0.32712,"date":"2025-06-02"}]}],"matchDetails":[{"type":"exact-direct-match","matcher":"python-matcher","searchedBy":{"language":"python","namespace":"github:language:python","package":{"name":"pillow","version":"9.0.1"}},"found":{"versionConstraint":"<9.2.0 (python)","vulnerabilityID":"GHSA-m2vv-5vj5-2hm7"},"fix":{"suggestedVersion":"9.2.0"}}],"artifact":{"id":"a191cb575c535611","name":"pillow","version":"9.0.1","type":"python","locations":[{"path":"/usr/lib/python3/dist-packages/Pillow-9.0.1.egg-info/PKG-INFO","accessPath":"/usr/lib/python3/dist-packages/Pillow-9.0.1.egg-info/PKG-INFO","annotations":{"evidence":"primary"}},{"path":"/usr/lib/python3/dist-packages/Pillow-9.0.1.egg-info/top_level.txt","accessPath":"/usr/lib/python3/dist-packages/Pillow-9.0.1.egg-info/top_level.txt","annotations":{"evidence":"supporting"}}],"language":"python","licenses":["HPND"],"cpes":["cpe:2.3:a:alex_clark_(pil_fork_author)_project:python-pillow:9.0.1:*:*:*:*:*:*:*","cpe:2.3:a:alex_clark_(pil_fork_author)_project:python_pillow:9.0.1:*:*:*:*:*:*:*","cpe:2.3:a:alex_clark_(pil_fork_author)project:python-pillow:9.0.1:*:*:*:*:*:*:*","cpe:2.3:a:alex_clark_(pil_fork_author)project:python_pillow:9.0.1:*:*:*:*:*:*:*","cpe:2.3:a:alex_clark_(pil_fork_author)_project:pillow:9.0.1:*:*:*:*:*:*:*","cpe:2.3:a:alex_clark_(pil_fork_author):python-pillow:9.0.1:*:*:*:*:*:*:*","cpe:2.3:a:alex_clark_(pil_fork_author):python_pillow:9.0.1:*:*:*:*:*:*:*","cpe:2.3:a:alex_clark_(pil_fork_author)project:pillow:9.0.1:*:*:*:*:*:*:*","cpe:2.3:a:alex_clark_(pil_fork_author):pillow:9.0.1:*:*:*:*:*:*:*","cpe:2.3:a:aclark_project:python-pillow:9.0.1:*:*:*:*:*:*:*","cpe:2.3:a:aclark_project:python_pillow:9.0.1:*:*:*:*:*:*:*","cpe:2.3:a:aclarkproject:python-pillow:9.0.1:*:*:*:*:*:*:*","cpe:2.3:a:aclarkproject:python_pillow:9.0.1:*:*:*:*:*:*:*","cpe:2.3:a:python-pillow:python-pillow:9.0.1:*:*:*:*:*:*:*","cpe:2.3:a:python-pillow:python_pillow:9.0.1:*:*:*:*:*:*:*","cpe:2.3:a:python_pillow:python-pillow:9.0.1:*:*:*:*:*:*:*","cpe:2.3:a:python_pillow:python_pillow:9.0.1:*:*:*:*:*:*:*","cpe:2.3:a:aclark_project:pillow:9.0.1:*:*:*:*:*:*:*","cpe:2.3:a:aclark:python-pillow:9.0.1:*:*:*:*:*:*:*","cpe:2.3:a:aclark:python_pillow:9.0.1:*:*:*:*:*:*:*","cpe:2.3:a:aclarkproject:pillow:9.0.1:*:*:*:*:*:*:*","cpe:2.3:a:pillow:python-pillow:9.0.1:*:*:*:*:*:*:*","cpe:2.3:a:pillow:python_pillow:9.0.1:*:*:*:*:*:*:*","cpe:2.3:a:python-pillow:pillow:9.0.1:*:*:*:*:*:*:*","cpe:2.3:a:python:python-pillow:9.0.1:*:*:*:*:*:*:*","cpe:2.3:a:python:python_pillow:9.0.1:*:*:*:*:*:*:*","cpe:2.3:a:python_pillow:pillow:9.0.1:*:*:*:*:*:*:*","cpe:2.3:a:aclark:pillow:9.0.1:*:*:*:*:*:*:*","cpe:2.3:a:pillow:pillow:9.0.1:*:*:*:*:*:*:*","cpe:2.3:a:python:pillow:9.0.1:*:*:*:*:*:*:*"],"purl":"pkg:pypi/pillow@9.0.1","upstreams":[]}},{"vulnerability":{"id":"GHSA-xqr8-7jwr-rhp7","dataSource":"https://github.com/advisories/GHSA-xqr8-7jwr-rhp7","namespace":"github:language:python","severity":"High","urls":[],"description":"Removal of e-Tugra root certificate","cvss":[{"type":"Secondary","version":"3.1","vector":"CVSS:3.1/AV:N/AC:L/PR:N/UI:N/S:U/C:N/I:H/A:N","metrics":{"baseScore":7.5,"exploitabilityScore":3.9,"impactScore":3.6},"vendorMetadata":{}}],"epss":[{"cve":"CVE-2023-37920","epss":0.00112,"percentile":0.30834,"date":"2025-06-02"}],"fix":{"versions":["2023.7.22"],"state":"fixed"},"advisories":[],"risk":0.08399999999999999},"relatedVulnerabilities":[{"id":"CVE-2023-37920","dataSource":"https://nvd.nist.gov/vuln/detail/CVE-2023-37920","namespace":"nvd:cpe","severity":"Critical","urls":["https://github.com/certifi/python-certifi/commit/8fb96ed81f71e7097ed11bc4d9b19afd7ea5c909","https://github.com/certifi/python-certifi/security/advisories/GHSA-xqr8-7jwr-rhp7","https://groups.google.com/a/mozilla.org/g/dev-security-policy/c/C-HrP1SEq1A","https://lists.fedoraproject.org/archives/list/package-announce@lists.fedoraproject.org/message/5EX6NG7WUFNUKGFHLM35KHHU3GAKXRTG/","https://github.com/certifi/python-certifi/commit/8fb96ed81f71e7097ed11bc4d9b19afd7ea5c909","https://github.com/certifi/python-certifi/security/advisories/GHSA-xqr8-7jwr-rhp7","https://groups.google.com/a/mozilla.org/g/dev-security-policy/c/C-HrP1SEq1A","https://lists.fedoraproject.org/archives/list/package-announce@lists.fedoraproject.org/message/5EX6NG7WUFNUKGFHLM35KHHU3GAKXRTG/","https://security.netapp.com/advisory/ntap-20240912-0002/"],"description":"Certifi is a curated collection of Root Certificates for validating the trustworthiness of SSL certificates while verifying the identity of TLS hosts. Certifi prior to version 2023.07.22 recognizes \"e-Tugra\" root certificates. e-Tugra's root certificates were subject to an investigation prompted by reporting of security issues in their systems. Certifi 2023.07.22 removes root certificates from \"e-Tugra\" from the root store.","cvss":[{"source":"nvd@nist.gov","type":"Primary","version":"3.1","vector":"CVSS:3.1/AV:N/AC:L/PR:N/UI:N/S:U/C:H/I:H/A:H","metrics":{"baseScore":9.8,"exploitabilityScore":3.9,"impactScore":5.9},"vendorMetadata":{}},{"source":"security-advisories@github.com","type":"Secondary","version":"3.1","vector":"CVSS:3.1/AV:N/AC:L/PR:N/UI:N/S:U/C:N/I:H/A:N","metrics":{"baseScore":7.5,"exploitabilityScore":3.9,"impactScore":3.6},"vendorMetadata":{}}],"epss":[{"cve":"CVE-2023-37920","epss":0.00112,"percentile":0.30834,"date":"2025-06-02"}]}],"matchDetails":[{"type":"exact-direct-match","matcher":"python-matcher","searchedBy":{"language":"python","namespace":"github:language:python","package":{"name":"certifi","version":"2020.6.20"}},"found":{"versionConstraint":">=2015.4.28,<2023.7.22 (python)","vulnerabilityID":"GHSA-xqr8-7jwr-rhp7"},"fix":{"suggestedVersion":"2023.7.22"}}],"artifact":{"id":"fcb59256428582cf","name":"certifi","version":"2020.6.20","type":"python","locations":[{"path":"/usr/lib/python3/dist-packages/certifi-2020.6.20.egg-info/PKG-INFO","accessPath":"/usr/lib/python3/dist-packages/certifi-2020.6.20.egg-info/PKG-INFO","annotations":{"evidence":"primary"}},{"path":"/usr/lib/python3/dist-packages/certifi-2020.6.20.egg-info/top_level.txt","accessPath":"/usr/lib/python3/dist-packages/certifi-2020.6.20.egg-info/top_level.txt","annotations":{"evidence":"supporting"}}],"language":"python","licenses":["MPL-2.0"],"cpes":["cpe:2.3:a:certifi:certifi:2020.6.20:*:*:*:*:python:*:*"],"purl":"pkg:pypi/certifi@2020.6.20","upstreams":[]}},{"vulnerability":{"id":"GHSA-44wm-f244-xhp3","dataSource":"https://github.com/advisories/GHSA-44wm-f244-xhp3","namespace":"github:language:python","severity":"High","urls":[],"description":"Pillow buffer overflow vulnerability","cvss":[{"type":"Secondary","version":"3.1","vector":"CVSS:3.1/AV:L/AC:H/PR:L/UI:R/S:U/C:H/I:H/A:H","metrics":{"baseScore":6.7,"exploitabilityScore":0.8,"impactScore":5.9},"vendorMetadata":{}}],"epss":[{"cve":"CVE-2024-28219","epss":0.00095,"percentile":0.27871,"date":"2025-06-02"}],"fix":{"versions":["10.3.0"],"state":"fixed"},"advisories":[],"risk":0.06745},"relatedVulnerabilities":[{"id":"CVE-2024-28219","dataSource":"https://nvd.nist.gov/vuln/detail/CVE-2024-28219","namespace":"nvd:cpe","severity":"Medium","urls":["https://lists.debian.org/debian-lts-announce/2024/04/msg00008.html","https://lists.fedoraproject.org/archives/list/package-announce%40lists.fedoraproject.org/message/4XLPUT3VK4GQ6EVY525TT2QNUIXNRU5M/","https://pillow.readthedocs.io/en/stable/releasenotes/10.3.0.html#security","https://lists.debian.org/debian-lts-announce/2024/04/msg00008.html","https://lists.fedoraproject.org/archives/list/package-announce%40lists.fedoraproject.org/message/4XLPUT3VK4GQ6EVY525TT2QNUIXNRU5M/","https://pillow.readthedocs.io/en/stable/releasenotes/10.3.0.html#security"],"description":"In _imagingcms.c in Pillow before 10.3.0, a buffer overflow exists because strcpy is used instead of strncpy.","cvss":[{"source":"cve@mitre.org","type":"Secondary","version":"3.1","vector":"CVSS:3.1/AV:L/AC:H/PR:L/UI:R/S:U/C:H/I:H/A:H","metrics":{"baseScore":6.7,"exploitabilityScore":0.8,"impactScore":5.9},"vendorMetadata":{}}],"epss":[{"cve":"CVE-2024-28219","epss":0.00095,"percentile":0.27871,"date":"2025-06-02"}]}],"matchDetails":[{"type":"exact-direct-match","matcher":"python-matcher","searchedBy":{"language":"python","namespace":"github:language:python","package":{"name":"pillow","version":"9.0.1"}},"found":{"versionConstraint":"<10.3.0 (python)","vulnerabilityID":"GHSA-44wm-f244-xhp3"},"fix":{"suggestedVersion":"10.3.0"}}],"artifact":{"id":"a191cb575c535611","name":"pillow","version":"9.0.1","type":"python","locations":[{"path":"/usr/lib/python3/dist-packages/Pillow-9.0.1.egg-info/PKG-INFO","accessPath":"/usr/lib/python3/dist-packages/Pillow-9.0.1.egg-info/PKG-INFO","annotations":{"evidence":"primary"}},{"path":"/usr/lib/python3/dist-packages/Pillow-9.0.1.egg-info/top_level.txt","accessPath":"/usr/lib/python3/dist-packages/Pillow-9.0.1.egg-info/top_level.txt","annotations":{"evidence":"supporting"}}],"language":"python","licenses":["HPND"],"cpes":["cpe:2.3:a:alex_clark_(pil_fork_author)_project:python-pillow:9.0.1:*:*:*:*:*:*:*","cpe:2.3:a:alex_clark_(pil_fork_author)_project:python_pillow:9.0.1:*:*:*:*:*:*:*","cpe:2.3:a:alex_clark_(pil_fork_author)project:python-pillow:9.0.1:*:*:*:*:*:*:*","cpe:2.3:a:alex_clark_(pil_fork_author)project:python_pillow:9.0.1:*:*:*:*:*:*:*","cpe:2.3:a:alex_clark_(pil_fork_author)_project:pillow:9.0.1:*:*:*:*:*:*:*","cpe:2.3:a:alex_clark_(pil_fork_author):python-pillow:9.0.1:*:*:*:*:*:*:*","cpe:2.3:a:alex_clark_(pil_fork_author):python_pillow:9.0.1:*:*:*:*:*:*:*","cpe:2.3:a:alex_clark_(pil_fork_author)project:pillow:9.0.1:*:*:*:*:*:*:*","cpe:2.3:a:alex_clark_(pil_fork_author):pillow:9.0.1:*:*:*:*:*:*:*","cpe:2.3:a:aclark_project:python-pillow:9.0.1:*:*:*:*:*:*:*","cpe:2.3:a:aclark_project:python_pillow:9.0.1:*:*:*:*:*:*:*","cpe:2.3:a:aclarkproject:python-pillow:9.0.1:*:*:*:*:*:*:*","cpe:2.3:a:aclarkproject:python_pillow:9.0.1:*:*:*:*:*:*:*","cpe:2.3:a:python-pillow:python-pillow:9.0.1:*:*:*:*:*:*:*","cpe:2.3:a:python-pillow:python_pillow:9.0.1:*:*:*:*:*:*:*","cpe:2.3:a:python_pillow:python-pillow:9.0.1:*:*:*:*:*:*:*","cpe:2.3:a:python_pillow:python_pillow:9.0.1:*:*:*:*:*:*:*","cpe:2.3:a:aclark_project:pillow:9.0.1:*:*:*:*:*:*:*","cpe:2.3:a:aclark:python-pillow:9.0.1:*:*:*:*:*:*:*","cpe:2.3:a:aclark:python_pillow:9.0.1:*:*:*:*:*:*:*","cpe:2.3:a:aclarkproject:pillow:9.0.1:*:*:*:*:*:*:*","cpe:2.3:a:pillow:python-pillow:9.0.1:*:*:*:*:*:*:*","cpe:2.3:a:pillow:python_pillow:9.0.1:*:*:*:*:*:*:*","cpe:2.3:a:python-pillow:pillow:9.0.1:*:*:*:*:*:*:*","cpe:2.3:a:python:python-pillow:9.0.1:*:*:*:*:*:*:*","cpe:2.3:a:python:python_pillow:9.0.1:*:*:*:*:*:*:*","cpe:2.3:a:python_pillow:pillow:9.0.1:*:*:*:*:*:*:*","cpe:2.3:a:aclark:pillow:9.0.1:*:*:*:*:*:*:*","cpe:2.3:a:pillow:pillow:9.0.1:*:*:*:*:*:*:*","cpe:2.3:a:python:pillow:9.0.1:*:*:*:*:*:*:*"],"purl":"pkg:pypi/pillow@9.0.1","upstreams":[]}},{"vulnerability":{"id":"GHSA-fpfv-jqm9-f5jm","dataSource":"https://github.com/advisories/GHSA-fpfv-jqm9-f5jm","namespace":"github:language:python","severity":"Medium","urls":[],"description":"Incorrect Comparison in NumPy","cvss":[{"type":"Secondary","version":"3.1","vector":"CVSS:3.1/AV:N/AC:L/PR:N/UI:N/S:U/C:N/I:N/A:L","metrics":{"baseScore":5.3,"exploitabilityScore":3.9,"impactScore":1.5},"vendorMetadata":{}}],"epss":[{"cve":"CVE-2021-34141","epss":0.00061,"percentile":0.19481,"date":"2025-06-02"}],"fix":{"versions":["1.22"],"state":"fixed"},"advisories":[],"risk":0.031415},"relatedVulnerabilities":[{"id":"CVE-2021-34141","dataSource":"https://nvd.nist.gov/vuln/detail/CVE-2021-34141","namespace":"nvd:cpe","severity":"Medium","urls":["https://github.com/numpy/numpy/issues/18993","https://www.oracle.com/security-alerts/cpujul2022.html","https://github.com/numpy/numpy/issues/18993","https://www.oracle.com/security-alerts/cpujul2022.html"],"description":"An incomplete string comparison in the numpy.core component in NumPy before 1.22.0 allows attackers to trigger slightly incorrect copying by constructing specific string objects. NOTE: the vendor states that this reported code behavior is \"completely harmless.\"","cvss":[{"source":"nvd@nist.gov","type":"Primary","version":"3.1","vector":"CVSS:3.1/AV:N/AC:L/PR:N/UI:N/S:U/C:N/I:N/A:L","metrics":{"baseScore":5.3,"exploitabilityScore":3.9,"impactScore":1.5},"vendorMetadata":{}},{"source":"nvd@nist.gov","type":"Primary","version":"2.0","vector":"AV:N/AC:L/Au:N/C:N/I:N/A:P","metrics":{"baseScore":5,"exploitabilityScore":10,"impactScore":2.9},"vendorMetadata":{}}],"epss":[{"cve":"CVE-2021-34141","epss":0.00061,"percentile":0.19481,"date":"2025-06-02"}]}],"matchDetails":[{"type":"exact-direct-match","matcher":"python-matcher","searchedBy":{"language":"python","namespace":"github:language:python","package":{"name":"numpy","version":"1.21.5"}},"found":{"versionConstraint":"<1.22 (python)","vulnerabilityID":"GHSA-fpfv-jqm9-f5jm"},"fix":{"suggestedVersion":"1.22"}}],"artifact":{"id":"747b8bd501c21bc7","name":"numpy","version":"1.21.5","type":"python","locations":[{"path":"/usr/lib/python3/dist-packages/numpy-1.21.5.egg-info/PKG-INFO","accessPath":"/usr/lib/python3/dist-packages/numpy-1.21.5.egg-info/PKG-INFO","annotations":{"evidence":"primary"}},{"path":"/usr/lib/python3/dist-packages/numpy-1.21.5.egg-info/top_level.txt","accessPath":"/usr/lib/python3/dist-packages/numpy-1.21.5.egg-info/top_level.txt","annotations":{"evidence":"supporting"}}],"language":"python","licenses":["BSD"],"cpes":["cpe:2.3:a:travis_e__oliphant_et_al__project:python-numpy:1.21.5:*:*:*:*:*:*:*","cpe:2.3:a:travis_e__oliphant_et_al__project:python_numpy:1.21.5:*:*:*:*:*:*:*","cpe:2.3:a:travis_e__oliphant_et_al_project:python-numpy:1.21.5:*:*:*:*:*:*:*","cpe:2.3:a:travis_e__oliphant_et_al_project:python_numpy:1.21.5:*:*:*:*:*:*:*","cpe:2.3:a:travis_e__oliphant_et_al__project:numpy:1.21.5:*:*:*:*:*:*:*","cpe:2.3:a:travis_e__oliphant_et_al_:python-numpy:1.21.5:*:*:*:*:*:*:*","cpe:2.3:a:travis_e__oliphant_et_al_:python_numpy:1.21.5:*:*:*:*:*:*:*","cpe:2.3:a:travis_e__oliphant_et_al_project:numpy:1.21.5:*:*:*:*:*:*:*","cpe:2.3:a:travis_e__oliphant_et_al_:numpy:1.21.5:*:*:*:*:*:*:*","cpe:2.3:a:python-numpy:python-numpy:1.21.5:*:*:*:*:*:*:*","cpe:2.3:a:python-numpy:python_numpy:1.21.5:*:*:*:*:*:*:*","cpe:2.3:a:python_numpy:python-numpy:1.21.5:*:*:*:*:*:*:*","cpe:2.3:a:python_numpy:python_numpy:1.21.5:*:*:*:*:*:*:*","cpe:2.3:a:python:python-numpy:1.21.5:*:*:*:*:*:*:*","cpe:2.3:a:python:python_numpy:1.21.5:*:*:*:*:*:*:*","cpe:2.3:a:numpy:python-numpy:1.21.5:*:*:*:*:*:*:*","cpe:2.3:a:numpy:python_numpy:1.21.5:*:*:*:*:*:*:*","cpe:2.3:a:python-numpy:numpy:1.21.5:*:*:*:*:*:*:*","cpe:2.3:a:python_numpy:numpy:1.21.5:*:*:*:*:*:*:*","cpe:2.3:a:python:numpy:1.21.5:*:*:*:*:*:*:*","cpe:2.3:a:numpy:numpy:1.21.5:*:*:*:*:*:*:*"],"purl":"pkg:pypi/numpy@1.21.5","upstreams":[]}},{"vulnerability":{"id":"GHSA-g4mx-q9vg-27p4","dataSource":"https://github.com/advisories/GHSA-g4mx-q9vg-27p4","namespace":"github:language:python","severity":"Medium","urls":[],"description":"urllib3's request body not stripped after redirect from 303 status changes request method to GET","cvss":[{"type":"Secondary","version":"3.1","vector":"CVSS:3.1/AV:A/AC:H/PR:H/UI:N/S:U/C:H/I:N/A:N","metrics":{"baseScore":4.2,"exploitabilityScore":0.6,"impactScore":3.6},"vendorMetadata":{}}],"epss":[{"cve":"CVE-2023-45803","epss":0.00062,"percentile":0.19668,"date":"2025-06-02"}],"fix":{"versions":["1.26.18"],"state":"fixed"},"advisories":[],"risk":0.02852},"relatedVulnerabilities":[{"id":"CVE-2023-45803","dataSource":"https://nvd.nist.gov/vuln/detail/CVE-2023-45803","namespace":"nvd:cpe","severity":"Medium","urls":["https://github.com/urllib3/urllib3/commit/4e98d57809dacab1cbe625fddeec1a290c478ea9","https://github.com/urllib3/urllib3/security/advisories/GHSA-g4mx-q9vg-27p4","https://lists.fedoraproject.org/archives/list/package-announce@lists.fedoraproject.org/message/4R2Y5XK3WALSR3FNAGN7JBYV2B343ZKB/","https://lists.fedoraproject.org/archives/list/package-announce@lists.fedoraproject.org/message/5F5CUBAN5XMEBVBZPHFITBLMJV5FIJJ5/","https://lists.fedoraproject.org/archives/list/package-announce@lists.fedoraproject.org/message/PPDPLM6UUMN55ESPQWJFLLIZY4ZKCNRX/","https://www.rfc-editor.org/rfc/rfc9110.html#name-get","https://github.com/urllib3/urllib3/commit/4e98d57809dacab1cbe625fddeec1a290c478ea9","https://github.com/urllib3/urllib3/security/advisories/GHSA-g4mx-q9vg-27p4","https://lists.fedoraproject.org/archives/list/package-announce@lists.fedoraproject.org/message/4R2Y5XK3WALSR3FNAGN7JBYV2B343ZKB/","https://lists.fedoraproject.org/archives/list/package-announce@lists.fedoraproject.org/message/5F5CUBAN5XMEBVBZPHFITBLMJV5FIJJ5/","https://lists.fedoraproject.org/archives/list/package-announce@lists.fedoraproject.org/message/PPDPLM6UUMN55ESPQWJFLLIZY4ZKCNRX/","https://www.rfc-editor.org/rfc/rfc9110.html#name-get"],"description":"urllib3 is a user-friendly HTTP client library for Python. urllib3 previously wouldn't remove the HTTP request body when an HTTP redirect response using status 301, 302, or 303 after the request had its method changed from one that could accept a request body (like `POST`) to `GET` as is required by HTTP RFCs. Although this behavior is not specified in the section for redirects, it can be inferred by piecing together information from different sections and we have observed the behavior in other major HTTP client implementations like curl and web browsers. Because the vulnerability requires a previously trusted service to become compromised in order to have an impact on confidentiality we believe the exploitability of this vulnerability is low. Additionally, many users aren't putting sensitive data in HTTP request bodies, if this is the case then this vulnerability isn't exploitable. Both of the following conditions must be true to be affected by this vulnerability: 1. Using urllib3 and submitting sensitive information in the HTTP request body (such as form data or JSON) and 2. The origin service is compromised and starts redirecting using 301, 302, or 303 to a malicious peer or the redirected-to service becomes compromised. This issue has been addressed in versions 1.26.18 and 2.0.7 and users are advised to update to resolve this issue. Users unable to update should disable redirects for services that aren't expecting to respond with redirects with `redirects=False` and disable automatic redirects with `redirects=False` and handle 301, 302, and 303 redirects manually by stripping the HTTP request body.","cvss":[{"source":"nvd@nist.gov","type":"Primary","version":"3.1","vector":"CVSS:3.1/AV:A/AC:H/PR:H/UI:N/S:U/C:H/I:N/A:N","metrics":{"baseScore":4.2,"exploitabilityScore":0.6,"impactScore":3.6},"vendorMetadata":{}},{"source":"security-advisories@github.com","type":"Secondary","version":"3.1","vector":"CVSS:3.1/AV:A/AC:H/PR:H/UI:N/S:U/C:H/I:N/A:N","metrics":{"baseScore":4.2,"exploitabilityScore":0.6,"impactScore":3.6},"vendorMetadata":{}}],"epss":[{"cve":"CVE-2023-45803","epss":0.00062,"percentile":0.19668,"date":"2025-06-02"}]}],"matchDetails":[{"type":"exact-direct-match","matcher":"python-matcher","searchedBy":{"language":"python","namespace":"github:language:python","package":{"name":"urllib3","version":"1.26.5"}},"found":{"versionConstraint":">=0,<1.26.18 (python)","vulnerabilityID":"GHSA-g4mx-q9vg-27p4"},"fix":{"suggestedVersion":"1.26.18"}}],"artifact":{"id":"043d634961733646","name":"urllib3","version":"1.26.5","type":"python","locations":[{"path":"/usr/lib/python3/dist-packages/urllib3-1.26.5.egg-info/PKG-INFO","accessPath":"/usr/lib/python3/dist-packages/urllib3-1.26.5.egg-info/PKG-INFO","annotations":{"evidence":"primary"}},{"path":"/usr/lib/python3/dist-packages/urllib3-1.26.5.egg-info/top_level.txt","accessPath":"/usr/lib/python3/dist-packages/urllib3-1.26.5.egg-info/top_level.txt","annotations":{"evidence":"supporting"}}],"language":"python","licenses":["MIT"],"cpes":["cpe:2.3:a:python:urllib3:1.26.5:*:*:*:*:*:*:*"],"purl":"pkg:pypi/urllib3@1.26.5","upstreams":[]}},{"vulnerability":{"id":"GHSA-43fp-rhv2-5gv8","dataSource":"https://github.com/advisories/GHSA-43fp-rhv2-5gv8","namespace":"github:language:python","severity":"Medium","urls":[],"description":"Certifi removing TrustCor root certificate","cvss":[{"type":"Secondary","version":"3.1","vector":"CVSS:3.1/AV:N/AC:L/PR:H/UI:N/S:C/C:N/I:H/A:N","metrics":{"baseScore":6.8,"exploitabilityScore":2.3,"impactScore":4},"vendorMetadata":{}}],"epss":[{"cve":"CVE-2022-23491","epss":0.00038,"percentile":0.10395,"date":"2025-06-02"}],"fix":{"versions":["2022.12.07"],"state":"fixed"},"advisories":[],"risk":0.02242},"relatedVulnerabilities":[{"id":"CVE-2022-23491","dataSource":"https://nvd.nist.gov/vuln/detail/CVE-2022-23491","namespace":"nvd:cpe","severity":"High","urls":["https://github.com/certifi/python-certifi/security/advisories/GHSA-43fp-rhv2-5gv8","https://groups.google.com/a/mozilla.org/g/dev-security-policy/c/oxX69KFvsm4/m/yLohoVqtCgAJ","https://github.com/certifi/python-certifi/security/advisories/GHSA-43fp-rhv2-5gv8","https://groups.google.com/a/mozilla.org/g/dev-security-policy/c/oxX69KFvsm4/m/yLohoVqtCgAJ","https://security.netapp.com/advisory/ntap-20230223-0010/"],"description":"Certifi is a curated collection of Root Certificates for validating the trustworthiness of SSL certificates while verifying the identity of TLS hosts. Certifi 2022.12.07 removes root certificates from \"TrustCor\" from the root store. These are in the process of being removed from Mozilla's trust store. TrustCor's root certificates are being removed pursuant to an investigation prompted by media reporting that TrustCor's ownership also operated a business that produced spyware. Conclusions of Mozilla's investigation can be found in the linked google group discussion.","cvss":[{"source":"nvd@nist.gov","type":"Primary","version":"3.1","vector":"CVSS:3.1/AV:N/AC:L/PR:N/UI:N/S:U/C:N/I:H/A:N","metrics":{"baseScore":7.5,"exploitabilityScore":3.9,"impactScore":3.6},"vendorMetadata":{}},{"source":"security-advisories@github.com","type":"Secondary","version":"3.1","vector":"CVSS:3.1/AV:N/AC:L/PR:H/UI:N/S:C/C:N/I:H/A:N","metrics":{"baseScore":6.8,"exploitabilityScore":2.3,"impactScore":4},"vendorMetadata":{}}],"epss":[{"cve":"CVE-2022-23491","epss":0.00038,"percentile":0.10395,"date":"2025-06-02"}]}],"matchDetails":[{"type":"exact-direct-match","matcher":"python-matcher","searchedBy":{"language":"python","namespace":"github:language:python","package":{"name":"certifi","version":"2020.6.20"}},"found":{"versionConstraint":">=2017.11.05,<2022.12.07 (python)","vulnerabilityID":"GHSA-43fp-rhv2-5gv8"},"fix":{"suggestedVersion":"2022.12.07"}}],"artifact":{"id":"fcb59256428582cf","name":"certifi","version":"2020.6.20","type":"python","locations":[{"path":"/usr/lib/python3/dist-packages/certifi-2020.6.20.egg-info/PKG-INFO","accessPath":"/usr/lib/python3/dist-packages/certifi-2020.6.20.egg-info/PKG-INFO","annotations":{"evidence":"primary"}},{"path":"/usr/lib/python3/dist-packages/certifi-2020.6.20.egg-info/top_level.txt","accessPath":"/usr/lib/python3/dist-packages/certifi-2020.6.20.egg-info/top_level.txt","annotations":{"evidence":"supporting"}}],"language":"python","licenses":["MPL-2.0"],"cpes":["cpe:2.3:a:certifi:certifi:2020.6.20:*:*:*:*:python:*:*"],"purl":"pkg:pypi/certifi@2020.6.20","upstreams":[]}},{"vulnerability":{"id":"GHSA-34jh-p97f-mpxf","dataSource":"https://github.com/advisories/GHSA-34jh-p97f-mpxf","namespace":"github:language:python","severity":"Medium","urls":[],"description":"urllib3's Proxy-Authorization request header isn't stripped during cross-origin redirects","cvss":[{"type":"Secondary","version":"3.1","vector":"CVSS:3.1/AV:N/AC:H/PR:H/UI:N/S:U/C:H/I:N/A:N","metrics":{"baseScore":4.4,"exploitabilityScore":0.8,"impactScore":3.6},"vendorMetadata":{}}],"epss":[{"cve":"CVE-2024-37891","epss":0.00033,"percentile":0.08074,"date":"2025-06-02"}],"fix":{"versions":["1.26.19"],"state":"fixed"},"advisories":[],"risk":0.01551},"relatedVulnerabilities":[{"id":"CVE-2024-37891","dataSource":"https://nvd.nist.gov/vuln/detail/CVE-2024-37891","namespace":"nvd:cpe","severity":"Medium","urls":["https://github.com/urllib3/urllib3/commit/accff72ecc2f6cf5a76d9570198a93ac7c90270e","https://github.com/urllib3/urllib3/security/advisories/GHSA-34jh-p97f-mpxf","https://github.com/urllib3/urllib3/commit/accff72ecc2f6cf5a76d9570198a93ac7c90270e","https://github.com/urllib3/urllib3/security/advisories/GHSA-34jh-p97f-mpxf","https://security.netapp.com/advisory/ntap-20240822-0003/","https://www.vicarius.io/vsociety/posts/proxy-authorization-header-handling-vulnerability-in-urllib3-cve-2024-37891"],"description":"urllib3 is a user-friendly HTTP client library for Python. When using urllib3's proxy support with `ProxyManager`, the `Proxy-Authorization` header is only sent to the configured proxy, as expected. However, when sending HTTP requests *without* using urllib3's proxy support, it's possible to accidentally configure the `Proxy-Authorization` header even though it won't have any effect as the request is not using a forwarding proxy or a tunneling proxy. In those cases, urllib3 doesn't treat the `Proxy-Authorization` HTTP header as one carrying authentication material and thus doesn't strip the header on cross-origin redirects. Because this is a highly unlikely scenario, we believe the severity of this vulnerability is low for almost all users. Out of an abundance of caution urllib3 will automatically strip the `Proxy-Authorization` header during cross-origin redirects to avoid the small chance that users are doing this on accident. Users should use urllib3's proxy support or disable automatic redirects to achieve safe processing of the `Proxy-Authorization` header, but we still decided to strip the header by default in order to further protect users who aren't using the correct approach. We believe the number of usages affected by this advisory is low. It requires all of the following to be true to be exploited: 1. Setting the `Proxy-Authorization` header without using urllib3's built-in proxy support. 2. Not disabling HTTP redirects. 3. Either not using an HTTPS origin server or for the proxy or target origin to redirect to a malicious origin. Users are advised to update to either version 1.26.19 or version 2.2.2. Users unable to upgrade may use the `Proxy-Authorization` header with urllib3's `ProxyManager`, disable HTTP redirects using `redirects=False` when sending requests, or not user the `Proxy-Authorization` header as mitigations.","cvss":[{"source":"security-advisories@github.com","type":"Secondary","version":"3.1","vector":"CVSS:3.1/AV:N/AC:H/PR:H/UI:N/S:U/C:H/I:N/A:N","metrics":{"baseScore":4.4,"exploitabilityScore":0.8,"impactScore":3.6},"vendorMetadata":{}}],"epss":[{"cve":"CVE-2024-37891","epss":0.00033,"percentile":0.08074,"date":"2025-06-02"}]}],"matchDetails":[{"type":"exact-direct-match","matcher":"python-matcher","searchedBy":{"language":"python","namespace":"github:language:python","package":{"name":"urllib3","version":"1.26.5"}},"found":{"versionConstraint":"<1.26.19 (python)","vulnerabilityID":"GHSA-34jh-p97f-mpxf"},"fix":{"suggestedVersion":"1.26.19"}}],"artifact":{"id":"043d634961733646","name":"urllib3","version":"1.26.5","type":"python","locations":[{"path":"/usr/lib/python3/dist-packages/urllib3-1.26.5.egg-info/PKG-INFO","accessPath":"/usr/lib/python3/dist-packages/urllib3-1.26.5.egg-info/PKG-INFO","annotations":{"evidence":"primary"}},{"path":"/usr/lib/python3/dist-packages/urllib3-1.26.5.egg-info/top_level.txt","accessPath":"/usr/lib/python3/dist-packages/urllib3-1.26.5.egg-info/top_level.txt","annotations":{"evidence":"supporting"}}],"language":"python","licenses":["MIT"],"cpes":["cpe:2.3:a:python:urllib3:1.26.5:*:*:*:*:*:*:*"],"purl":"pkg:pypi/urllib3@1.26.5","upstreams":[]}},{"vulnerability":{"id":"GHSA-9wx4-h78v-vm56","dataSource":"https://github.com/advisories/GHSA-9wx4-h78v-vm56","namespace":"github:language:python","severity":"Medium","urls":[],"description":"Requests `Session` object does not verify requests after making first request with verify=False","cvss":[{"type":"Secondary","version":"3.1","vector":"CVSS:3.1/AV:L/AC:H/PR:H/UI:R/S:U/C:H/I:H/A:N","metrics":{"baseScore":5.6,"exploitabilityScore":0.4,"impactScore":5.2},"vendorMetadata":{}}],"epss":[{"cve":"CVE-2024-35195","epss":0.00009,"percentile":0.0055,"date":"2025-06-02"}],"fix":{"versions":["2.32.0"],"state":"fixed"},"advisories":[],"risk":0.004770000000000001},"relatedVulnerabilities":[{"id":"CVE-2024-35195","dataSource":"https://nvd.nist.gov/vuln/detail/CVE-2024-35195","namespace":"nvd:cpe","severity":"Medium","urls":["https://github.com/psf/requests/commit/a58d7f2ffb4d00b46dca2d70a3932a0b37e22fac","https://github.com/psf/requests/pull/6655","https://github.com/psf/requests/security/advisories/GHSA-9wx4-h78v-vm56","https://lists.fedoraproject.org/archives/list/package-announce@lists.fedoraproject.org/message/IYLSNK5TL46Q6XPRVMHVWS63MVJQOK4Q/","https://lists.fedoraproject.org/archives/list/package-announce@lists.fedoraproject.org/message/N7WP6EYDSUOCOJYHDK5NX43PYZ4SNHGZ/","https://github.com/psf/requests/commit/a58d7f2ffb4d00b46dca2d70a3932a0b37e22fac","https://github.com/psf/requests/pull/6655","https://github.com/psf/requests/security/advisories/GHSA-9wx4-h78v-vm56","https://lists.fedoraproject.org/archives/list/package-announce@lists.fedoraproject.org/message/IYLSNK5TL46Q6XPRVMHVWS63MVJQOK4Q/","https://lists.fedoraproject.org/archives/list/package-announce@lists.fedoraproject.org/message/N7WP6EYDSUOCOJYHDK5NX43PYZ4SNHGZ/"],"description":"Requests is a HTTP library. Prior to 2.32.0, when making requests through a Requests `Session`, if the first request is made with `verify=False` to disable cert verification, all subsequent requests to the same host will continue to ignore cert verification regardless of changes to the value of `verify`. This behavior will continue for the lifecycle of the connection in the connection pool. This vulnerability is fixed in 2.32.0.","cvss":[{"source":"security-advisories@github.com","type":"Secondary","version":"3.1","vector":"CVSS:3.1/AV:L/AC:H/PR:H/UI:R/S:U/C:H/I:H/A:N","metrics":{"baseScore":5.6,"exploitabilityScore":0.4,"impactScore":5.2},"vendorMetadata":{}}],"epss":[{"cve":"CVE-2024-35195","epss":0.00009,"percentile":0.0055,"date":"2025-06-02"}]}],"matchDetails":[{"type":"exact-direct-match","matcher":"python-matcher","searchedBy":{"language":"python","namespace":"github:language:python","package":{"name":"requests","version":"2.25.1"}},"found":{"versionConstraint":"<2.32.0 (python)","vulnerabilityID":"GHSA-9wx4-h78v-vm56"},"fix":{"suggestedVersion":"2.32.0"}}],"artifact":{"id":"52e214e010c57a74","name":"requests","version":"2.25.1","type":"python","locations":[{"path":"/usr/lib/python3/dist-packages/requests-2.25.1.egg-info/PKG-INFO","accessPath":"/usr/lib/python3/dist-packages/requests-2.25.1.egg-info/PKG-INFO","annotations":{"evidence":"primary"}},{"path":"/usr/lib/python3/dist-packages/requests-2.25.1.egg-info/top_level.txt","accessPath":"/usr/lib/python3/dist-packages/requests-2.25.1.egg-info/top_level.txt","annotations":{"evidence":"supporting"}}],"language":"python","licenses":["Apache 2.0"],"cpes":["cpe:2.3:a:python:requests:2.25.1:*:*:*:*:*:*:*"],"purl":"pkg:pypi/requests@2.25.1","upstreams":[]}}],"source":{"type":"sbom-file","target":"chirp-snap-syft.json"},"distro":{"name":"","version":"","idLike":null},"descriptor":{"name":"grype","version":"0.92.2","configuration":{"output":["json=/Users/alan/Personal/popey/syft-snap-source-test/grype/v1.26.0/2025-05-21/chirp-snap-grype.json","template=/Users/alan/Personal/popey/syft-snap-source-test/grype/v1.26.0/2025-05-21/chirp-snap-grype.html"],"file":"","pretty":false,"distro":"","add-cpes-if-none":false,"output-template-file":"../html.tmpl","check-for-app-update":true,"only-fixed":false,"only-notfixed":false,"ignore-wontfix":"","platform":"","search":{"scope":"squashed","unindexed-archives":false,"indexed-archives":true},"ignore":[{"vulnerability":"","reason":"","namespace":"","fix-state":"","package":{"name":"kernel-headers","version":"","language":"","type":"rpm","location":"","upstream-name":"kernel"},"vex-status":"","vex-justification":"","match-type":"exact-indirect-match"},{"vulnerability":"","reason":"","namespace":"","fix-state":"","package":{"name":"linux(-.*)?-headers-.*","version":"","language":"","type":"deb","location":"","upstream-name":"linux.*"},"vex-status":"","vex-justification":"","match-type":"exact-indirect-match"},{"vulnerability":"","reason":"","namespace":"","fix-state":"","package":{"name":"linux-libc-dev","version":"","language":"","type":"deb","location":"","upstream-name":"linux"},"vex-status":"","vex-justification":"","match-type":"exact-indirect-match"}],"exclude":[],"externalSources":{"enable":false,"maven":{"searchUpstreamBySha1":true,"baseUrl":"https://search.maven.org/solrsearch/select","rateLimit":300000000}},"match":{"java":{"using-cpes":false},"jvm":{"using-cpes":true},"dotnet":{"using-cpes":false},"golang":{"using-cpes":false,"always-use-cpe-for-stdlib":true,"allow-main-module-pseudo-version-comparison":false},"javascript":{"using-cpes":false},"python":{"using-cpes":false},"ruby":{"using-cpes":false},"rust":{"using-cpes":false},"stock":{"using-cpes":true}},"fail-on-severity":"","registry":{"insecure-skip-tls-verify":false,"insecure-use-http":false,"auth":null,"ca-cert":""},"show-suppressed":false,"by-cve":false,"SortBy":{"sort-by":"risk"},"name":"","default-image-pull-source":"","vex-documents":[],"vex-add":[],"match-upstream-kernel-headers":false,"db":{"cache-dir":"/Users/alan/Library/Caches/grype/db","update-url":"https://grype.anchore.io/databases","ca-cert":"","auto-update":true,"validate-by-hash-on-start":true,"validate-age":true,"max-allowed-built-age":432000000000000,"require-update-check":false,"update-available-timeout":30000000000,"update-download-timeout":300000000000,"max-update-check-frequency":7200000000000},"exp":{},"dev":{"db":{"debug":false}}},"db":{"status":{"schemaVersion":"v6.0.2","from":"https://grype.anchore.io/databases/v6/vulnerability-db_v6.0.2_2025-06-03T01:32:03Z_1748923962.tar.zst?checksum=sha256%3A02a6127217518a93e0ef8739cb9f2d659109fff0d9cc25f0c13d5ca8b307a118","built":"2025-06-03T04:12:42Z","path":"/Users/alan/Library/Caches/grype/db/6/vulnerability.db","valid":true},"providers":{"alpine":{"captured":"2025-06-03T01:32:27Z","input":"xxh64:a6a065ece392f6d1"},"amazon":{"captured":"2025-06-03T01:32:10Z","input":"xxh64:b65d458d6858d72a"},"chainguard":{"captured":"2025-06-03T01:32:15Z","input":"xxh64:bad092583f78d4dd"},"debian":{"captured":"2025-06-03T01:32:22Z","input":"xxh64:b4717cad509f50d8"},"epss":{"captured":"2025-06-03T01:32:43Z","input":"xxh64:2f9660be834e0ce6"},"github":{"captured":"2025-06-03T01:32:21Z","input":"xxh64:cdd94f1952fcf607"},"kev":{"captured":"2025-06-03T01:32:03Z","input":"xxh64:575844c3434cc70b"},"mariner":{"captured":"2025-06-03T01:32:11Z","input":"xxh64:7628f1adb0b9b12b"},"nvd":{"captured":"2025-06-03T01:32:30Z","input":"xxh64:463c4e9a1d401101"},"oracle":{"captured":"2025-06-03T01:32:27Z","input":"xxh64:257df3059d28d823"},"rhel":{"captured":"2025-06-03T01:33:04Z","input":"xxh64:59e7d70bf674afdc"},"sles":{"captured":"2025-06-03T01:32:46Z","input":"xxh64:3cf7f372ea0c981a"},"ubuntu":{"captured":"2025-06-03T01:33:20Z","input":"xxh64:075be8afc080f2be"},"wolfi":{"captured":"2025-06-03T01:32:31Z","input":"xxh64:1aa0b26b004bdd38"}}},"timestamp":"2025-06-03T13:02:07.416595+01:00"}}
